---
title: 'R package:blupADC-Feature 1'
author: qsmei
date: '2021-04-14'
slug: blupADC
categories: ["Data format"]
tags: []
subtitle: ''
summary: 'Convert genotype data format'
authors: []
lastmod: '2021-04-14T14:59:41+08:00'
featured: no
image:
  caption: ''
  focal_point: ''
  preview_only: no
projects: []
output:
  blogdown::html_page:
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(collapse = TRUE)
```

## Overview 

`genotype_data_format_conversion` is the basic function of  package:`blupADC`.  By applying `genotype_data_format_conversion` , we can convert multiple genotype data formats in an easy way, including Hapmap, Plink, Blupf90, Numeric and VCF.  

**ðŸ‘‰ Note:  For analyzing VCF format data,  `blupADC` has encapsulated software Plink `version-1.9` and  VCFtools `version-0.1.17`. If you want change this version, you could set the related parameters in the part of Advanced parameter.**

## Example

```R
library(blupADC)
sum_data=genotype_data_format_conversion(
         input_data_hmp=data_hmp,  #provided hapmap data object 
         output_data_type=c("Plink","Blupf90","Numeric"),# output data format
         return_result = TRUE,      # return result 
         cpu_cores=1                # number of cpu 
         )
```

```{r, echo=FALSE, message=FALSE, warning=FALSE}
library(blupADC)
colnames(data_hmp)[1]="rs#"
data_hmp[1:6,1]=paste0("SNP",1:6)
sum_data=genotype_data_format_conversion(
         input_data_hmp=data_hmp[1:6,],  #provided hapmap data object 
         output_data_type=c("Plink","Blupf90","Numeric"),# output data format
         return_result = TRUE,      # return result 
         cpu_cores=1                # number of cpu 
                  )
```

## Parameter 

### ðŸ¤¡Primary  

-   **1: input_data_plink_ped**

User-provided  ped format genotype data ï¼Œ`data.frame` classã€‚

```{r input_data_plink_map2,echo=FALSE}
data_ped=sum_data$ped
data_map=sum_data$map
colnames(data_map)=NULL
data_genumeric=sum_data$genumeric
data_blupf90=sum_data$blupf90
colnames(data_blupf90)=NULL
knitr::kable(head(data_ped[,1:10]))
```

-   **2:input_data_plink_map**

User-provided  map format genotype data ï¼Œ`data.frame` classã€‚

```{r input_data_plink_map,echo=FALSE}
knitr::kable(head(data_map))
```

-   **3:input_data_hmp**

User-provided  hapmap format genotype data ï¼Œ`data.frame` classã€‚

```{r input_data_hmp,echo=FALSE}
knitr::kable(head(data_hmp[,1:15]))
```

-   **4:input_data_blupf90**

User-provided  BLUPF90 format genotype data ï¼Œ`data.frame` classã€‚

```{r input_data_blupf90,echo=FALSE}
knitr::kable(head(data_blupf90))
```

-   **5:input_data_genumeric**

User-provided  Numeric format genotype data ï¼Œ`matrix(numeric)` classã€‚

```{r input_data_genumeric,echo=FALSE}
knitr::kable(head(data_genumeric[,1:6]))
```

**Note:** rownames stand for individualsï¼Œcolnames stands for SNPs.  `input_data_genumeric`  should contain both rownames and colnames.

In addition, for convenience, users can provide the  file name, file path, and file type of genotype data directly without reading them in R enviroument by themself .

- **6:input_data_type**

  User-provided  file type of genotype data ï¼Œ`character` class.

  -   Hapmap
  -   Plink
  -   Blupf90
  -   Numeric
  -   VCF

-   **7:input_data_path**

User-provided  file path of genotype data ï¼Œ`character` class.

-   **8:input_data_name**

User-provided  file name of genotype data ï¼Œ`character` class.

**Note:**  if input_data_type is Plink or VCF, user don't need to include suffix in the file name of genotype data. 

eg. for Plink type data, files name are test1.map and test1.pedï¼Œwe should set`input_data_name="test1"`ã€‚

-   **9:return_result**

If return result, `logical` class. Default is FALSE.

Additionally,  for convenience, users can  save output genotype data  into local computer .

-   **10:output_data_path**

File path of output genotype data, `character` class.

-   **11:output_data_name**

File name of output genotype data, `character` class.

-   **12:output_data_type**

    File type of output genotype data, `character` class.

    -   Hapmap
    -   Plink
    -   Blupf90
    -   Numeric
    -   VCF

**Note**: users can  output multiple formats of genotype data simultaneously. e.g.`output_data_type=c("Hapmap","Plink","Blupf90","Numeric")`, outout 4 types of genotype data simultaneously ã€‚

### ðŸ’¨Secondary

-   **13:cpu_cores**

Number of cpu in calculating,  `numeric`calss. Default is 1.

-   **14:miss_base**

Missing genotype character,  `character`calss. Default is "NN".

-   **15:miss_base_num** 

Missing genotype number after numeric conversion,  `numeric`calss. Default is 5.

-   **16:plink_software_path**

Path of plink software in user computer. `character`calss.

-   **17:plink_software_name**

Name of plink software in user computer. `character`calss.

-   **18:vcftools_software_path**

Path of vcftools software in user computer. `character`calss.

-   **19:vcftools_software_name**

Name of vcftools software in user computer. `character`calss.
