---
title: 'R package:blupADC-Feature 1'
author: qsmei
date: '2021-04-14'
slug: blupADC
categories: ["Data format"]
tags: []
subtitle: ''
summary: 'Convert genotype data format'
authors: []
lastmod: '2021-04-14T14:59:41+08:00'
featured: no
image:
  caption: ''
  focal_point: ''
  preview_only: no
projects: []
output:
  blogdown::html_page:
    toc: true
---

<script src="{{< blogdown/postref >}}index_files/header-attrs/header-attrs.js"></script>

<div id="TOC">
<ul>
<li><a href="#overview">Overview</a></li>
<li><a href="#example">Example</a></li>
<li><a href="#parameter">Parameter</a>
<ul>
<li><a href="#primary">🤡Primary</a></li>
<li><a href="#secondary">💨Secondary</a></li>
</ul></li>
</ul>
</div>

<div id="overview" class="section level2">
<h2>Overview</h2>
<p><code>genotype_data_format_conversion</code> is the basic function of package:<code>blupADC</code>. By applying <code>genotype_data_format_conversion</code> , we can convert multiple genotype data formats in an easy way, including Hapmap, Plink, Blupf90, Numeric and VCF.</p>
<p><strong>👉 Note: For analyzing VCF format data, <code>blupADC</code> has encapsulated software Plink <code>version-1.9</code> and VCFtools <code>version-0.1.17</code>. If you want change this version, you could set the related parameters in the part of Advanced parameter.</strong></p>
</div>
<div id="example" class="section level2">
<h2>Example</h2>
<pre class="r"><code>library(blupADC)
sum_data=genotype_data_format_conversion(
         input_data_hmp=data_hmp,  #provided hapmap data object 
         output_data_type=c(&quot;Plink&quot;,&quot;Blupf90&quot;,&quot;Numeric&quot;),# output data format
         return_result = TRUE,      # return result 
         cpu_cores=1                # number of cpu 
         )</code></pre>
<pre><code>## Start numeric(0,1,2) format conversion...... 
## Complete Blupf90  format conversion 
## Complete Plink Data Format Conversion</code></pre>
</div>
<div id="parameter" class="section level2">
<h2>Parameter</h2>
<div id="primary" class="section level3">
<h3>🤡Primary</h3>
<ul>
<li><strong>1: input_data_plink_ped</strong></li>
</ul>
<p>User-provided ped format genotype data ，<code>data.frame</code> class。</p>
<table>
<tbody>
<tr class="odd">
<td align="left">3098</td>
<td align="left">3098</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">C</td>
<td align="left">C</td>
<td align="left">G</td>
<td align="left">G</td>
</tr>
<tr class="even">
<td align="left">3498</td>
<td align="left">3498</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">A</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="left">G</td>
</tr>
<tr class="odd">
<td align="left">3297</td>
<td align="left">3297</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">A</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="left">G</td>
</tr>
<tr class="even">
<td align="left">2452</td>
<td align="left">2452</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">C</td>
<td align="left">C</td>
<td align="left">G</td>
<td align="left">G</td>
</tr>
<tr class="odd">
<td align="left">4255</td>
<td align="left">4255</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">A</td>
<td align="left">C</td>
<td align="left">G</td>
<td align="left">G</td>
</tr>
<tr class="even">
<td align="left">2946</td>
<td align="left">2946</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">C</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="left">G</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>2:input_data_plink_map</strong></li>
</ul>
<p>User-provided map format genotype data ，<code>data.frame</code> class。</p>
<table>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="left">SNP1</td>
<td align="right">0.224488</td>
<td align="right">224488</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">SNP2</td>
<td align="right">0.293696</td>
<td align="right">293696</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">SNP3</td>
<td align="right">0.333333</td>
<td align="right">333333</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">SNP4</td>
<td align="right">0.464830</td>
<td align="right">464830</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">SNP5</td>
<td align="right">0.722623</td>
<td align="right">722623</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">SNP6</td>
<td align="right">0.838596</td>
<td align="right">838596</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>3:input_data_hmp</strong></li>
</ul>
<p>User-provided hapmap format genotype data ，<code>data.frame</code> class。</p>
<table style="width:100%;">
<colgroup>
<col width="4%" />
<col width="7%" />
<col width="5%" />
<col width="6%" />
<col width="6%" />
<col width="8%" />
<col width="6%" />
<col width="8%" />
<col width="9%" />
<col width="9%" />
<col width="6%" />
<col width="4%" />
<col width="4%" />
<col width="4%" />
<col width="4%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">rs#</th>
<th align="left">alleles</th>
<th align="right">chrom</th>
<th align="right">pos</th>
<th align="left">strand</th>
<th align="left">assembly</th>
<th align="left">center</th>
<th align="left">protLSID</th>
<th align="left">assayLSID</th>
<th align="left">panelLSID</th>
<th align="left">QCcode</th>
<th align="left">3098</th>
<th align="left">3498</th>
<th align="left">3297</th>
<th align="left">2452</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">SNP1</td>
<td align="left">NA</td>
<td align="right">1</td>
<td align="right">224488</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">CC</td>
<td align="left">AC</td>
<td align="left">AC</td>
<td align="left">CC</td>
</tr>
<tr class="even">
<td align="left">SNP2</td>
<td align="left">NA</td>
<td align="right">1</td>
<td align="right">293696</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">GG</td>
<td align="left">TG</td>
<td align="left">TG</td>
<td align="left">GG</td>
</tr>
<tr class="odd">
<td align="left">SNP3</td>
<td align="left">NA</td>
<td align="right">1</td>
<td align="right">333333</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">GG</td>
<td align="left">TT</td>
<td align="left">TT</td>
<td align="left">GG</td>
</tr>
<tr class="even">
<td align="left">SNP4</td>
<td align="left">NA</td>
<td align="right">1</td>
<td align="right">464830</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">CC</td>
<td align="left">CC</td>
<td align="left">CC</td>
<td align="left">CC</td>
</tr>
<tr class="odd">
<td align="left">SNP5</td>
<td align="left">NA</td>
<td align="right">1</td>
<td align="right">722623</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">AA</td>
<td align="left">GG</td>
<td align="left">GG</td>
<td align="left">AA</td>
</tr>
<tr class="even">
<td align="left">SNP6</td>
<td align="left">NA</td>
<td align="right">1</td>
<td align="right">838596</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">CC</td>
<td align="left">TC</td>
<td align="left">TT</td>
<td align="left">CC</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>4:input_data_blupf90</strong></li>
</ul>
<p>User-provided BLUPF90 format genotype data ，<code>data.frame</code> class。</p>
<table>
<tbody>
<tr class="odd">
<td align="left">3098</td>
<td align="left">200000</td>
</tr>
<tr class="even">
<td align="left">3498</td>
<td align="left">112021</td>
</tr>
<tr class="odd">
<td align="left">3297</td>
<td align="left">112022</td>
</tr>
<tr class="even">
<td align="left">2452</td>
<td align="left">200000</td>
</tr>
<tr class="odd">
<td align="left">4255</td>
<td align="left">102011</td>
</tr>
<tr class="even">
<td align="left">2946</td>
<td align="left">212000</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>5:input_data_genumeric</strong></li>
</ul>
<p>User-provided Numeric format genotype data ，<code>matrix(numeric)</code> class。</p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">SNP1</th>
<th align="right">SNP2</th>
<th align="right">SNP3</th>
<th align="right">SNP4</th>
<th align="right">SNP5</th>
<th align="right">SNP6</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">3098</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">3498</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">3297</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">2452</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">4255</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">2946</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p><strong>Note:</strong> rownames stand for individuals，colnames stands for SNPs. <code>input_data_genumeric</code> should contain both rownames and colnames.</p>
<p>In addition, for convenience, users can provide the file name, file path, and file type of genotype data directly without reading them in R enviroument by themself .</p>
<ul>
<li><p><strong>6:input_data_type</strong></p>
<p>User-provided file type of genotype data ，<code>character</code> class.</p>
<ul>
<li>Hapmap</li>
<li>Plink</li>
<li>Blupf90</li>
<li>Numeric</li>
<li>VCF</li>
</ul></li>
<li><p><strong>7:input_data_path</strong></p></li>
</ul>
<p>User-provided file path of genotype data ，<code>character</code> class.</p>
<ul>
<li><strong>8:input_data_name</strong></li>
</ul>
<p>User-provided file name of genotype data ，<code>character</code> class.</p>
<p><strong>Note:</strong> if input_data_type is Plink or VCF, user don’t need to include suffix in the file name of genotype data.</p>
<p>eg. for Plink type data, files name are test1.map and test1.ped，we should set<code>input_data_name="test1"</code>。</p>
<ul>
<li><strong>9:return_result</strong></li>
</ul>
<p>If return result, <code>logical</code> class. Default is FALSE.</p>
<p>Additionally, for convenience, users can save output genotype data into local computer .</p>
<ul>
<li><strong>10:output_data_path</strong></li>
</ul>
<p>File path of output genotype data, <code>character</code> class.</p>
<ul>
<li><strong>11:output_data_name</strong></li>
</ul>
<p>File name of output genotype data, <code>character</code> class.</p>
<ul>
<li><p><strong>12:output_data_type</strong></p>
<p>File type of output genotype data, <code>character</code> class.</p>
<ul>
<li>Hapmap</li>
<li>Plink</li>
<li>Blupf90</li>
<li>Numeric</li>
<li>VCF</li>
</ul></li>
</ul>
<p><strong>Note</strong>: users can output multiple formats of genotype data simultaneously. e.g.<code>output_data_type=c("Hapmap","Plink","Blupf90","Numeric")</code>, outout 4 types of genotype data simultaneously 。</p>
</div>
<div id="secondary" class="section level3">
<h3>💨Secondary</h3>
<ul>
<li><strong>13:cpu_cores</strong></li>
</ul>
<p>Number of cpu in calculating, <code>numeric</code>calss. Default is 1.</p>
<ul>
<li><strong>14:miss_base</strong></li>
</ul>
<p>Missing genotype character, <code>character</code>calss. Default is “NN”.</p>
<ul>
<li><strong>15:miss_base_num</strong></li>
</ul>
<p>Missing genotype number after numeric conversion, <code>numeric</code>calss. Default is 5.</p>
<ul>
<li><strong>16:plink_software_path</strong></li>
</ul>
<p>Path of plink software in user computer. <code>character</code>calss.</p>
<ul>
<li><strong>17:plink_software_name</strong></li>
</ul>
<p>Name of plink software in user computer. <code>character</code>calss.</p>
<ul>
<li><strong>18:vcftools_software_path</strong></li>
</ul>
<p>Path of vcftools software in user computer. <code>character</code>calss.</p>
<ul>
<li><strong>19:vcftools_software_name</strong></li>
</ul>
<p>Name of vcftools software in user computer. <code>character</code>calss.</p>
</div>
</div>
