---
title: 'R package:blupADC-Feature 4'
author: qsmei
date: '2021-04-17'
slug: qc_imputation
categories:
  - Imputaion
  - QC
tags: []
subtitle: ''
summary: 'data qc and imputation'
authors: []
lastmod: '2021-04-17T21:07:25+08:00'
featured: no
image:
  caption: ''
  focal_point: ''
  preview_only: no
projects: []
toc: true
---

{{<toc>}}

## Overview

ðŸ¤– Generally, most genotype data need to perform quality control and imputation.  For convenience,  package `blupADC` provides `genotype_data_QC_Imputation` function to perform quality control and imputation by interfacing with  software **Plink** and software **Beagle** in an easy way (we only need to provide the software path and software name).   

**ðŸ‘‰ Note:  For convenience,  `blupADC` has encapsulated software Plink(for quality control)  `version-1.9` and software Beagle(for imputation) `version-5.2`. If you want change this version, you could set the related parameters in the part of Advanced parameter.**

## Example

``` {.R}
library(blupADC)
genotype_data_QC_Imputation(
            input_data_hmp=data_hmp,    #provided hapmap data object
            data_analysis_method="QC_Imputation",   #analysis method type,QC + imputatoin
            output_data_path="/root/result",        #output data path
            output_data_name="YY_data",             #output data name
            output_data_type="Plink"                #output data format 
            )                       
```

In the process of quality control and imputationï¼Œwe should provide genotype dataï¼Œthese parameters are the as in`genotype_data_format_conversion`function(see more details).

## Parameter

### ðŸ’™Basic

- **1ï¼šdata_analysis_method**

  Method of analyzing data, `character` class.

  -   `"QC"` : only perform quality control
  -   `"Imputation"` : only perform imputation
-   `"QC_Imputation"` : first perform quality control, and then  perform imputation


-   **2ï¼šqc_snp_rate**

Threshold of SNPs call rate in quality control, `numeric` class. Default is 0.1

ä½¿ç”¨Plinkè¿›è¡Œè´¨æŽ§æ—¶ï¼ŒSNP call rate çš„é˜ˆå€¼ï¼Œ`numeric`ç±»åž‹ï¼Œé»˜è®¤ä¸º0.1ã€‚

-   **3ï¼šqc_ind_rate**

Threshold of individuals call rate in quality control, `numeric` class. Default is 0.1

-   **4ï¼šqc_maf **  

Threshold of minor allele frequency(MAF)  in quality control, `numeric` class. Default is 0.05

-   **5ï¼šqc_hwe **   

Threshold of hardy weinberg equilibrium(HWE)  in quality control, `numeric` class. Default is 1e-7

### ðŸ’œAdvanced

-   **6ï¼šplink_software_path**

Path of software **Plink**  , `character` class.

-   **7ï¼šplink_software_name**

Name of  software  **Plink** , `character` class.

-   **8ï¼šbeagle_software_path**

Path of software **Beagle** , `character` class.

-   **9ï¼šbeagle_software_name**

Name of software **Beagle** , `character` class.

-   **10ï¼šbeagle_ref_data_path**  

File path of reference data in imputation, `character` class.

-   **11ï¼šbeagle_ref_data_name**

File name of reference data in imputation, `character` class.

-   **12ï¼šbeagle_ped_path**

File path of pedigree data in imputation, `character` class.

-   **13ï¼šbeagle_ped_name**

File name of pedigree data in imputation, `character` class.
