<!DOCTYPE html><html lang="en-us" >

<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  
  
  
    <meta name="generator" content="Wowchemy 5.1.0 for Hugo" />
  

  

  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Quanshun Mei" />

  
  
  
    
  
  <meta name="description" content="Genomic data format conversion" />

  
    <link rel="alternate" hreflang="zh" href="/zh/post/blupadc/" />
  
  <link rel="alternate" hreflang="en-us" href="/post/blupadc/" />

  







  




  
  
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
  

  
  
  
    <meta name="theme-color" content="#1565c0" />
  

  
  

  
  
  
  
    
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.14.0/css/all.min.css" integrity="sha256-FMvZuGapsJLjouA6k7Eo2lusoAX9i0ShlWFG6qt7SLc=" crossorigin="anonymous">

    
    

    
    
    
      
    
    
      
      
        
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/styles/github.min.css" crossorigin="anonymous" title="hl-light" media="print" onload="this.media='all'">
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/styles/dracula.min.css" crossorigin="anonymous" title="hl-dark" media="print" onload="this.media='all'" disabled>
        
      
    

    
    
    
      <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.7.1/leaflet.min.css" integrity="sha512-1xoFisiGdy9nvho8EgXuXvnpR5GAMSjFwp40gSRE3NwdUdIMIKuPa7bqoUhLD0O/5tPNhteAsE5XyyMi5reQVA==" crossorigin="anonymous" media="print" onload="this.media='all'">
    

    

    
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      
        
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
  

  
  
  
    
      
      <link rel="preload" as="style" href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&family=Roboto+Mono&family=Roboto:wght@400;700&display=swap">
      <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&family=Roboto+Mono&family=Roboto:wght@400;700&display=swap" media="print" onload="this.media='all'">
    
  

  
  
  
  
  
  <link rel="stylesheet" href="/css/wowchemy.9a66e344a68eb664b392d406a3f80726.css" />

  



  

  

  




  
  
  

  

  
    <link rel="manifest" href="/index.webmanifest" />
  

  <link rel="icon" type="image/png" href="/media/icon_hua2ec155b4296a9c9791d015323e16eb5_11927_32x32_fill_lanczos_center_2.png" />
  <link rel="apple-touch-icon" type="image/png" href="/media/icon_hua2ec155b4296a9c9791d015323e16eb5_11927_180x180_fill_lanczos_center_2.png" />

  <link rel="canonical" href="/post/blupadc/" />

  
  
  
  
  
  
  
  
    
    
  
  
  <meta property="twitter:card" content="summary" />
  
  <meta property="og:site_name" content="Quanshun Mei-Personal" />
  <meta property="og:url" content="/post/blupadc/" />
  <meta property="og:title" content="R package:blupADC-Feature 1 | Quanshun Mei-Personal" />
  <meta property="og:description" content="Genomic data format conversion" /><meta property="og:image" content="/media/icon_hua2ec155b4296a9c9791d015323e16eb5_11927_512x512_fill_lanczos_center_2.png" />
    <meta property="twitter:image" content="/media/icon_hua2ec155b4296a9c9791d015323e16eb5_11927_512x512_fill_lanczos_center_2.png" /><meta property="og:locale" content="en-us" />
  
    
      <meta
        property="article:published_time"
        content="2021-04-14T00:00:00&#43;00:00"
      />
    
    <meta property="article:modified_time" content="2021-04-14T14:59:41&#43;08:00">
  

  


    






  




<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "/post/blupadc/"
  },
  "headline": "R package:blupADC-Feature 1",
  
  "datePublished": "2021-04-14T00:00:00Z",
  "dateModified": "2021-04-14T14:59:41+08:00",
  
  "author": {
    "@type": "Person",
    "name": "Quanshun Mei"
  },
  
  "publisher": {
    "@type": "Organization",
    "name": "Quanshun Mei-Personal",
    "logo": {
      "@type": "ImageObject",
      "url": "/media/icon_hua2ec155b4296a9c9791d015323e16eb5_11927_192x192_fill_lanczos_center_2.png"
    }
  },
  "description": "Genomic data format conversion"
}
</script>

  

  

  

  





  <title>R package:blupADC-Feature 1 | Quanshun Mei-Personal</title>
</head>


<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" class="page-wrapper   " data-wc-page-id="b107928fdee05958ff3b5aa1f0ccc19d" >

  
  
  
  
  
  
  
  
  <script src="/js/wowchemy-init.min.226a9011996d125bf3fe4a5f22353a49.js"></script>

  


<aside class="search-modal" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#" aria-label="Close"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search" class="form-control"
        aria-label="Search...">
        
      </div>

      
      

      

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>



  <div class="page-header">
    












<nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id="navbar-main">
  <div class="container-xl">

    
    <div class="d-none d-lg-inline-flex">
      <a class="navbar-brand" href="/">Quanshun Mei-Personal</a>
    </div>
    

    
    <button type="button" class="navbar-toggler" data-toggle="collapse"
            data-target="#navbar-content" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
    <span><i class="fas fa-bars"></i></span>
    </button>
    

    
    <div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none">
      <a class="navbar-brand" href="/">Quanshun Mei-Personal</a>
    </div>
    

    
    
    <div class="navbar-collapse main-menu-item collapse justify-content-start" id="navbar-content">

      
      <ul class="navbar-nav d-md-inline-flex">
        

        

        
        
        
          
        

        

        
        
        
        

        
          
            
            
          
          
            
            
            
              
            
            
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#about"><span>About Me</span></a>
        </li>

        
        

        

        
        
        
          
        

        

        
        
        
        

        
          
            
            
          
          
            
            
            
              
            
            
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#projects"><span>Software</span></a>
        </li>

        
        

        

        
        
        
          
        

        

        
        
        
        

        
          
            
            
          
          
            
            
            
              
            
            
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#publications"><span>Publications</span></a>
        </li>

        
        

        

        
        
        
          
        

        

        
        
        
        

        
          
            
            
          
          
            
            
            
              
            
            
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#posts"><span>Posts</span></a>
        </li>

        
        

      

        
      </ul>
    </div>

    <ul class="nav-icons navbar-nav flex-row ml-auto d-flex pl-md-2">

      
      
        
          
          <li class="nav-item d-none d-lg-inline-flex">
            <a class="nav-link" href="/"  aria-label="">
              <i class="fas fa-" aria-hidden="true"></i>
            </a>
          </li>
        
          
          <li class="nav-item d-none d-lg-inline-flex">
            <a class="nav-link" href="/"  aria-label="">
              <i class="fas fa-" aria-hidden="true"></i>
            </a>
          </li>
        
          
          <li class="nav-item d-none d-lg-inline-flex">
            <a class="nav-link" href="/"  aria-label="">
              <i class="fas fa-" aria-hidden="true"></i>
            </a>
          </li>
        
      

      
      
      <li class="nav-item">
        <a class="nav-link js-search" href="#" aria-label="Search"><i class="fas fa-search" aria-hidden="true"></i></a>
      </li>
      

      
      
      <li class="nav-item dropdown theme-dropdown">
        <a href="#" class="nav-link" data-toggle="dropdown" aria-haspopup="true" aria-label="Display preferences">
          <i class="fas fa-moon" aria-hidden="true"></i>
        </a>
        <div class="dropdown-menu">
          <a href="#" class="dropdown-item js-set-theme-light">
            <span>Light</span>
          </a>
          <a href="#" class="dropdown-item js-set-theme-dark">
            <span>Dark</span>
          </a>
          <a href="#" class="dropdown-item js-set-theme-auto">
            <span>Automatic</span>
          </a>
        </div>
      </li>
      

      
      
      <li class="nav-item dropdown i18n-dropdown">
        <a href="#" class="nav-link " data-toggle="dropdown"
           aria-haspopup="true" aria-label="Languages">
          <i class="fas fa-globe mr-1" aria-hidden="true"></i></a>
        <div class="dropdown-menu">
          <div class="dropdown-item dropdown-item-active">
            <span>English</span>
          </div>
          
          <a class="dropdown-item" href="/zh/post/blupadc/">
            <span>中文 (简体)</span>
          </a>
          
        </div>
      </li>
      

    </ul>

  </div>
</nav>


  </div>

  <div class="page-body">
    <article class="article">

  












  

  
  
  
<div class="article-container pt-3">
  <h1>R package:blupADC-Feature 1</h1>

  

  
    


<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
          Last updated on
      
    
    Apr 14, 2021
  </span>
  

  

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    6 min read
  </span>
  

  
  
  
  
  
  

  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fas fa-folder mr-1"></i><a href="/category/data-format/">Data format</a></span>
  

</div>

    





  
</div>



  <div class="article-container">

    <div class="article-style">
      
<script src="/post/blupadc/index_files/header-attrs/header-attrs.js"></script>

<div id="TOC">
<ul>
<li><a href="#overview">Overview</a></li>
<li><a href="#example">Example</a></li>
<li><a href="#output">Output</a></li>
<li><a href="#parameter">Parameter</a>
<ul>
<li><a href="#basic">🤡Basic</a></li>
<li><a href="#advanced">💨Advanced</a></li>
</ul></li>
</ul>
</div>

<div id="overview" class="section level2">
<h2>Overview</h2>
<p><code>geno_format</code> is the basic function of package:<code>blupADC</code>. By applying <code>geno_format</code> , we can convert multiple genotype data formats in an easy way, including Hapmap, Plink, BLUPF90, Numeric, Haplotype and VCF.</p>
</div>
<div id="example" class="section level2">
<h2>Example</h2>
<div id="format-conversion-based-on-provided-r-variable" class="section level4">
<h4>Format conversion based on provided R variable</h4>
<pre class="r"><code>library(blupADC)
format_result=geno_format(
         input_data_hmp=example_data_hmp,  #provided hapmap data object 
         output_data_type=c(&quot;Plink&quot;,&quot;BLUPF90&quot;,&quot;Numeric&quot;,&quot;VCF&quot;),# output data format
         return_result = TRUE,      # return result 
         cpu_cores=1                # number of cpu 
         )</code></pre>
</div>
<div id="format-conversion-based-on-provided-data-path-and-data-name" class="section level4">
<h4>Format conversion based on provided data path and data name</h4>
<pre class="r"><code>#convert phased VCF data to haplotype format and  haplotype-based numeric format
library(blupADC)
data_path=system.file(&quot;extdata&quot;, package = &quot;blupADC&quot;)  #  path of example files 
phased_result=geno_format(
         input_data_path=data_path,      # input data path 
         input_data_name=&quot;example.vcf&quot;,  # input data name,for vcf data 
         input_data_type=&quot;VCF&quot;,          # input data type
         phased_genotype=TRUE,           # whether the vcf data has been phased
         haplotype_window_nSNP=5,        # according to nSNP define block,
         output_data_type=c(&quot;Haplotype&quot;,&quot;Numeric&quot;),# output data format
         return_result=TRUE,             #save result as a R environment variable
         cpu_cores=1                     # number of cpu 
                  )</code></pre>
</div>
<div id="format-conversion-via-bigmemory-method" class="section level4">
<h4>Format conversion via bigmemory method</h4>
<pre class="r"><code>library(blupADC)
data_path=system.file(&quot;extdata&quot;, package = &quot;blupADC&quot;)  #  path of example files 
phased=geno_format(
         input_data_path=data_path,      # input data path 
         input_data_name=&quot;example.vcf&quot;,  # input data name,for vcf data
         input_data_type=&quot;VCF&quot;,          # input data type
         phased_genotype=TRUE,           # whether the vcf data has been phased
         haplotype_window_nSNP=5,        # according to nSNP define haplotype-block,
         bigmemory_cal=TRUE,             # format conversion via bigmemory object
         bigmemory_data_path=getwd(),    # path of bigmemory data 
         bigmemory_data_name=&quot;test_blupADC&quot;, #name of bigmemory data 
         output_data_type=c(&quot;Haplotype&quot;,&quot;Numeric&quot;),# output data format
         return_result=TRUE,             #save result in R environment
         cpu_cores=1                     # number of cpu 
                  )</code></pre>
</div>
</div>
<div id="output" class="section level2">
<h2>Output</h2>
<p>According to the result of output, we find that the output contains 5 parts, including:</p>
<ul>
<li><strong>hmp </strong> : <code>Hapmap</code> format genotype data</li>
</ul>
<p>The first column stands for the name of SNP, the thrid column stands for chromosome, the fourth column stands for the physical postion, and the twelth column and the after columns stand for the genotype data</p>
<table style="width:100%;">
<colgroup>
<col width="4%" />
<col width="7%" />
<col width="5%" />
<col width="6%" />
<col width="6%" />
<col width="8%" />
<col width="6%" />
<col width="8%" />
<col width="10%" />
<col width="10%" />
<col width="6%" />
<col width="4%" />
<col width="4%" />
<col width="4%" />
<col width="4%" />
</colgroup>
<thead>
<tr class="header">
<th>rs#</th>
<th>alleles</th>
<th>chrom</th>
<th>pos</th>
<th>strand</th>
<th>assembly</th>
<th>center</th>
<th>protLSID</th>
<th>assayLSID</th>
<th>panelLSID</th>
<th>QCcode</th>
<th>3098</th>
<th>3498</th>
<th>3297</th>
<th>2452</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>SNP1</td>
<td>NA</td>
<td>1</td>
<td>224488</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>CC</td>
<td>AC</td>
<td>AC</td>
<td>CC</td>
</tr>
<tr class="even">
<td>SNP2</td>
<td>NA</td>
<td>1</td>
<td>293696</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>GG</td>
<td>TG</td>
<td>TG</td>
<td>GG</td>
</tr>
<tr class="odd">
<td>SNP3</td>
<td>NA</td>
<td>1</td>
<td>333333</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>GG</td>
<td>TT</td>
<td>TT</td>
<td>GG</td>
</tr>
<tr class="even">
<td>SNP4</td>
<td>NA</td>
<td>1</td>
<td>464830</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>CC</td>
<td>CC</td>
<td>CC</td>
<td>CC</td>
</tr>
<tr class="odd">
<td>SNP5</td>
<td>NA</td>
<td>1</td>
<td>722623</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>AA</td>
<td>GG</td>
<td>GG</td>
<td>AA</td>
</tr>
<tr class="even">
<td>SNP6</td>
<td>NA</td>
<td>1</td>
<td>838596</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>CC</td>
<td>TC</td>
<td>TT</td>
<td>CC</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>ped</strong> : <code>Plink</code> format ped data</li>
</ul>
<p>The first column stands for family name,the second column stands for the individual name,the seventh column and the after columns stand for the genotype data</p>
<table>
<thead>
<tr class="header">
<th>3098</th>
<th>3098</th>
<th>0</th>
<th>0</th>
<th>0</th>
<th>0</th>
<th>C</th>
<th>C</th>
<th>G</th>
<th>G</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>3498</td>
<td>3498</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>A</td>
<td>C</td>
<td>T</td>
<td>G</td>
</tr>
<tr class="even">
<td>3297</td>
<td>3297</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>A</td>
<td>C</td>
<td>T</td>
<td>G</td>
</tr>
<tr class="odd">
<td>2452</td>
<td>2452</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>C</td>
<td>C</td>
<td>G</td>
<td>G</td>
</tr>
<tr class="even">
<td>4255</td>
<td>4255</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>A</td>
<td>C</td>
<td>G</td>
<td>G</td>
</tr>
<tr class="odd">
<td>2946</td>
<td>2946</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>C</td>
<td>C</td>
<td>T</td>
<td>G</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>map </strong>: <code>Plink</code>format map data</li>
</ul>
<p>The first column stands for chromosome, the second column stands for the name of SNP, the thrid column stands for the genetic positon(CM), and the fourth column stands for the physical position</p>
<table>
<thead>
<tr class="header">
<th>1</th>
<th>SNP1</th>
<th>0.224488</th>
<th>224488</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>SNP2</td>
<td>0.293696</td>
<td>293696</td>
</tr>
<tr class="even">
<td>1</td>
<td>SNP3</td>
<td>0.333333</td>
<td>333333</td>
</tr>
<tr class="odd">
<td>1</td>
<td>SNP4</td>
<td>0.464830</td>
<td>464830</td>
</tr>
<tr class="even">
<td>1</td>
<td>SNP5</td>
<td>0.722623</td>
<td>722623</td>
</tr>
<tr class="odd">
<td>1</td>
<td>SNP6</td>
<td>0.838596</td>
<td>838596</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>blupf90 </strong>: <code>BLUPF90</code> format genotype data</li>
</ul>
<p>The first column stands for individual name, the second column stands for the genotype data(numeric)</p>
<table>
<thead>
<tr class="header">
<th>3098</th>
<th>200000</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>3498</td>
<td>112021</td>
</tr>
<tr class="even">
<td>3297</td>
<td>112022</td>
</tr>
<tr class="odd">
<td>2452</td>
<td>200000</td>
</tr>
<tr class="even">
<td>4255</td>
<td>102011</td>
</tr>
<tr class="odd">
<td>2946</td>
<td>212000</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>numeric </strong>: <code>Numeric</code> format genotype data</li>
</ul>
<p>rownames of numeric data stands for the individual name, colnames of numeric data stands for the name of SNP, 0,1,2 stand for the numeric genotype</p>
<table>
<thead>
<tr class="header">
<th>2</th>
<th>0</th>
<th>0</th>
<th>0</th>
<th>0</th>
<th>0</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>1</td>
<td>2</td>
<td>0</td>
<td>2</td>
<td>1</td>
</tr>
<tr class="even">
<td>1</td>
<td>1</td>
<td>2</td>
<td>0</td>
<td>2</td>
<td>2</td>
</tr>
<tr class="odd">
<td>2</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>1</td>
<td>0</td>
<td>2</td>
<td>0</td>
<td>1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>2</td>
<td>1</td>
<td>2</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
</tbody>
</table>
<ul>
<li><p><strong>haplotype_hap</strong>: <code>Haplotype</code> format genotype data.</p>
<p>Row stands for marker, column stands for individual, each individual has two columns;</p>
<table>
<thead>
<tr class="header">
<th>0</th>
<th>0</th>
<th>0</th>
<th>1</th>
<th>1</th>
<th>0</th>
<th>0</th>
<th>0</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>1</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>0</td>
<td>0</td>
</tr>
</tbody>
</table></li>
<li><p><strong>haplotype_sample</strong>: <code>Haplotype</code> format genotype data</p>
<table>
<thead>
<tr class="header">
<th>3098</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>3498</td>
</tr>
<tr class="even">
<td>3297</td>
</tr>
<tr class="odd">
<td>2452</td>
</tr>
<tr class="even">
<td>4255</td>
</tr>
<tr class="odd">
<td>2946</td>
</tr>
</tbody>
</table></li>
<li><p><strong>haplotype_map</strong>: <code>Haplotype</code> format genotype data</p>
<table>
<thead>
<tr class="header">
<th>1</th>
<th>SNP1</th>
<th>224488</th>
<th>C</th>
<th>A</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>SNP2</td>
<td>293696</td>
<td>G</td>
<td>T</td>
</tr>
<tr class="even">
<td>1</td>
<td>SNP3</td>
<td>333333</td>
<td>T</td>
<td>G</td>
</tr>
<tr class="odd">
<td>1</td>
<td>SNP4</td>
<td>464830</td>
<td>A</td>
<td>G</td>
</tr>
<tr class="even">
<td>1</td>
<td>SNP5</td>
<td>722623</td>
<td>C</td>
<td>T</td>
</tr>
<tr class="odd">
<td>1</td>
<td>SNP6</td>
<td>838596</td>
<td>C</td>
<td>A</td>
</tr>
</tbody>
</table></li>
<li><p><strong>vcf </strong>: <code>VCF</code> format genotype data</p></li>
</ul>
<table>
<thead>
<tr class="header">
<th>##fileformat=VCFv4.2</th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>##source=“beagle.29May21.d6d.jar”</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>##INFO&lt;ID=AF,Number=A,Type=Float&gt;</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>##INFO&lt;ID=IMP,Number=0,Type=Flag"&gt;</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>##FORMAT&lt;ID=GT,Number=1,Type=String&gt;</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>#CHROM</td>
<td>POS</td>
<td>ID</td>
<td>REF</td>
<td>ALT</td>
<td>QUAL</td>
<td>FILTER</td>
<td>INFO</td>
<td>FORMAT</td>
<td>3498</td>
<td>3297</td>
</tr>
<tr class="even">
<td>1</td>
<td>6260</td>
<td>M2</td>
<td>T</td>
<td>A</td>
<td>.</td>
<td>PASS</td>
<td>.</td>
<td>GT</td>
<td>1|0</td>
<td>0|1</td>
</tr>
<tr class="odd">
<td>1</td>
<td>15289</td>
<td>M17</td>
<td>A</td>
<td>T</td>
<td>.</td>
<td>PASS</td>
<td>.</td>
<td>GT</td>
<td>0|0</td>
<td>0|0</td>
</tr>
</tbody>
</table>
</div>
<div id="parameter" class="section level2">
<h2>Parameter</h2>
<div id="basic" class="section level3">
<h3>🤡Basic</h3>
<ul>
<li><strong>1: input_data_plink_ped</strong></li>
</ul>
<p>User-provided <code>Plink</code>-ped format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<ul>
<li><strong>2:input_data_plink_map</strong></li>
</ul>
<p>User-provided <code>Plink</code>-map format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<ul>
<li><strong>3:input_data_hmp</strong></li>
</ul>
<p>User-provided <code>Hapmap</code> format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<ul>
<li><strong>4:input_data_BLUPF90</strong></li>
</ul>
<p>User-provided <code>BLUPF90</code> format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<ul>
<li><strong>5:input_data_numeric</strong></li>
</ul>
<p>User-provided <code>Numeric</code> format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<ul>
<li><strong>6:input_data_haplotype_hap</strong></li>
</ul>
<p>User-provided <code>Haplotype</code> format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<ul>
<li><strong>7:input_data_haplotype_sample</strong></li>
</ul>
<p>User-provided <code>Haplotype</code> format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<ul>
<li><strong>8:input_data_haplotype_map</strong></li>
</ul>
<p>User-provided <code>Haplotype</code> format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<ul>
<li><strong>9:input_data_vcf</strong></li>
</ul>
<p>User-provided <code>VCF</code> format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<p><strong>Note:</strong> <code>input_data_numeric</code> should contain both rownames and colnames.</p>
<p><strong>In addition, for convenience, users can provide the file name, file path, and file type of genotype data directly without reading them in R environment.</strong></p>
<ul>
<li><p><strong>10:input_data_type</strong></p>
<p>File type of provided genotype data ，<code>character</code> class.</p>
<ul>
<li>Hapmap</li>
<li>Plink</li>
<li>BLUPF90</li>
<li>Numeric</li>
<li>Haplotype</li>
<li>VCF</li>
</ul></li>
<li><p><strong>11:input_data_path</strong></p></li>
</ul>
<p>File path of provided genotype data ，<code>character</code> class.</p>
<ul>
<li><strong>12:input_data_name</strong></li>
</ul>
<p>File name of provided genotype data ，<code>character</code> class.</p>
<p><strong>Note:</strong> if input_data_type is Plink or VCF, user don’t need to include suffix in the file name of genotype data.</p>
<p>eg. for Plink type data, files name are test1.map and test1.ped，we should set<code>input_data_name="test1"</code>.</p>
<ul>
<li><strong>13:output_data_name</strong></li>
</ul>
<p>File name of output genotype data, <code>character</code> class.</p>
<ul>
<li><p><strong>14:output_data_type</strong></p>
<p>File type of output genotype data, <code>character</code> class.</p>
<ul>
<li>Hapmap</li>
<li>Plink</li>
<li>BLUPF90</li>
<li>Numeric</li>
<li>Haplotype</li>
<li>VCF</li>
</ul></li>
</ul>
<p><strong>Note</strong>: users can output multiple formats of genotype data simultaneously. e.g.<code>output_data_type=c("Hapmap","Plink","BLUPF90","Numeric")</code>, outout 4 types of genotype data simultaneously .</p>
<ul>
<li><strong>15:return_result</strong></li>
</ul>
<p>Whether return result, <code>logical</code> class. Default is FALSE.</p>
<p>Additionally, for convenience, users can save output genotype data into local computer .</p>
<ul>
<li><strong>16:bigmemory_cal</strong></li>
</ul>
<p>Whether using bigmemory method to calculate. <code>logical</code> class. Default is FALSE.</p>
<ul>
<li><strong>17:bigmemory_data_path</strong></li>
</ul>
<p>The file path bigmemory data . <code>character</code> class.</p>
<ul>
<li><strong>18:bigmemory_data_name</strong></li>
</ul>
<p>The file name bigmemory data . <code>character</code> class.</p>
<ul>
<li><strong>19:phased_genotype</strong></li>
</ul>
<p>Whether genotype data has been phased. <code>logical</code> class. Default is FALSE.</p>
<ul>
<li><strong>20:haplotype_window_nSNP</strong></li>
</ul>
<p>According to the number of consecutive SNPs define haplotype block. <code>numeric</code> class. Default is NULL.</p>
<ul>
<li><strong>21:haplotype_window_kb</strong></li>
</ul>
<p>According to the physical location define haplotype block. <code>numeric</code> class. Default is NULL.</p>
<ul>
<li><strong>22:haplotype_window_block</strong></li>
</ul>
<p>According to user-provided block to define haplotype block . <code>numeric</code> class. Default is NULL.</p>
<p>The first column is the position of window start, the second column is the position of window end.</p>
<table>
<thead>
<tr class="header">
<th>1</th>
<th>5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>6</td>
<td>10</td>
</tr>
<tr class="even">
<td>11</td>
<td>15</td>
</tr>
<tr class="odd">
<td>16</td>
<td>20</td>
</tr>
<tr class="even">
<td>21</td>
<td>25</td>
</tr>
<tr class="odd">
<td>26</td>
<td>30</td>
</tr>
</tbody>
</table>
</div>
<div id="advanced" class="section level3">
<h3>💨Advanced</h3>
<ul>
<li><strong>23:cpu_cores</strong></li>
</ul>
<p>Number of cpu in calculating, <code>numeric</code> class. Default is 1.</p>
<ul>
<li><strong>24:miss_base</strong></li>
</ul>
<p>Missing genotype character, <code>character</code> class. Default is “NN”.</p>
<ul>
<li><strong>25:miss_base_num</strong></li>
</ul>
<p>Missing genotype number after numeric conversion, <code>numeric</code> class. Default is 5.</p>
<ul>
<li><strong>26:miss_base_num</strong></li>
</ul>
<p>Missing genotype number after numeric conversion, <code>numeric</code> class. Default is 5.</p>
</div>
</div>

    </div>

    








<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=/post/blupadc/&amp;text=R%20package:blupADC-Feature%201" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=/post/blupadc/&amp;t=R%20package:blupADC-Feature%201" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=R%20package:blupADC-Feature%201&amp;body=/post/blupadc/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=/post/blupadc/&amp;title=R%20package:blupADC-Feature%201" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="whatsapp://send?text=R%20package:blupADC-Feature%201%20/post/blupadc/" target="_blank" rel="noopener" class="share-btn-whatsapp">
          <i class="fab fa-whatsapp"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://service.weibo.com/share/share.php?url=/post/blupadc/&amp;title=R%20package:blupADC-Feature%201" target="_blank" rel="noopener" class="share-btn-weibo">
          <i class="fab fa-weibo"></i>
        </a>
      </li>
    
  </ul>
</div>











  
  



  
  
  
    
  
  
  
  <div class="media author-card content-widget-hr">
    
      
      <img class="avatar mr-3 avatar-circle" src="/author/quanshun-mei/avatar_hua28df167aad214296f4568b85755d250_255914_270x270_fill_q75_lanczos_center.JPG" alt="Quanshun Mei">
    

    <div class="media-body">
      <h5 class="card-title">Quanshun Mei</h5>
      <h6 class="card-subtitle">PhD student in Quantitative Genetics of Animal Breeding</h6>
      <p class="card-text">My research interests include applying genomic selection and machine learning in animal breeding.</p>
      <ul class="network-icon" aria-hidden="true">
  
    
    
    
      
    
    
    
    
    
    <li>
      <a href="mailto:quanshun1994@gmail.com" >
        <i class="fas fa-envelope"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="/" >
        <i class="fas fa-"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://scholar.google.com/citations?user=yMQEkysAAAAJ&amp;hl=en&amp;oi=ao" target="_blank" rel="noopener">
        <i class="fas fa-graduation-cap"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="/" >
        <i class="fas fa-"></i>
      </a>
    </li>
  
    
    
    
    
    
    
    
      
    
    <li>
      <a href="https://github.com/qsmei" target="_blank" rel="noopener">
        <i class="fabi am curious to learn and try new thingsfa fabi am curious to learn and try new thingsfa-github"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="/" >
        <i class="fas fa-"></i>
      </a>
    </li>
  
</ul>

    </div>
  </div>
















  
  





  </div>
</article>
  </div>

  <div class="page-footer">
    
    
    <div class="container">
      <footer class="site-footer">

  

  

  <p class="powered-by">
    © {2021} Qsmei
  </p>

  
  






  <p class="powered-by">
    
    
    
      
      
      
      
      
      
      Published with <a href="https://wowchemy.com/?utm_campaign=poweredby" target="_blank" rel="noopener">Wowchemy</a> — the free, <a href="https://github.com/wowchemy/wowchemy-hugo-modules" target="_blank" rel="noopener">open source</a> website builder that empowers creators.
    
  </p>
</footer>

    </div>
    
  </div>

  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

      

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha256-9/aliU8dGd2tb6OSsuzixeV4y/faTqgFtohetphbbj0=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/instant.page/5.1.0/instantpage.min.js" integrity="sha512-1+qUtKoh9XZW7j+6LhRMAyOrgSQKenQ4mluTR+cvxXjP1Z54RxZuzstR/H9kgPXQsVB8IW7DMDFUJpzLjvhGSQ==" crossorigin="anonymous"></script>

      
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js" integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin="anonymous"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js" integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin="anonymous"></script>
      

      
      

      

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/highlight.min.js" integrity="sha512-TDKKr+IvoqZnPzc3l35hdjpHD0m+b2EC2SrLEgKDRWpxf2rFCxemkgvJ5kfU48ip+Y+m2XVKyOCD85ybtlZDmw==" crossorigin="anonymous"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/languages/r.min.js"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/languages/python.min.js"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/languages/latex.min.js"></script>
        
      

    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.7.1/leaflet.min.js" integrity="sha512-SeiQaaDh73yrb56sTW/RgVdi/mMqNeM2oBwubFHagc5BkixSpP1fvqF47mKzPGWYSSy4RwbBunrJBQ4Co8fRWA==" crossorigin="anonymous"></script>
    

    
    

    
    
    
      
      <script id="search-hit-fuse-template" type="text/x-template">
        <div class="search-hit" id="summary-{{key}}">
          <div class="search-hit-content">
            <div class="search-hit-name">
              <a href="{{relpermalink}}">{{title}}</a>
              <div class="article-metadata search-hit-type">{{type}}</div>
              <p class="search-hit-description">{{snippet}}</p>
            </div>
          </div>
        </div>
      </script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
      
    

    
    

    
    
    
    

    
    <script src="/js/bootstrap.bundle.min.6aed84840afc03ab4d5750157f69c120.js"></script>

    
    
      
      
      
      
      
      
      
    

    
    
    
    
    
    
    
    
      
      
    
    
    <script src="/en/js/wowchemy.min.b61a8f62b6e5c0cd322c8158c5b5dfb6.js"></script>

    
  <script async defer src="https://buttons.github.io/buttons.js"></script>




</body>
</html>
