<!DOCTYPE html><html lang="en-us" >

<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  
  
  
    <meta name="generator" content="Wowchemy 5.1.0 for Hugo" />
  

  

  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Mei Quanshun" />

  
  
  
    
  
  <meta name="description" content="Convert genotype data format" />

  
  <link rel="alternate" hreflang="en-us" href="/post/blupadc/" />

  







  




  
  
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
  

  
  
  
    <meta name="theme-color" content="#1565c0" />
  

  
  

  
  
  
  
    
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.14.0/css/all.min.css" integrity="sha256-FMvZuGapsJLjouA6k7Eo2lusoAX9i0ShlWFG6qt7SLc=" crossorigin="anonymous">

    
    

    
    
    
      
    
    
      
      
        
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/styles/github.min.css" crossorigin="anonymous" title="hl-light" media="print" onload="this.media='all'">
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/styles/dracula.min.css" crossorigin="anonymous" title="hl-dark" media="print" onload="this.media='all'" disabled>
        
      
    

    
    
    
      <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.7.1/leaflet.min.css" integrity="sha512-1xoFisiGdy9nvho8EgXuXvnpR5GAMSjFwp40gSRE3NwdUdIMIKuPa7bqoUhLD0O/5tPNhteAsE5XyyMi5reQVA==" crossorigin="anonymous" media="print" onload="this.media='all'">
    

    

    
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      
        
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
  

  
  
  
    
      
      <link rel="preload" as="style" href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&family=Roboto+Mono&family=Roboto:wght@400;700&display=swap">
      <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&family=Roboto+Mono&family=Roboto:wght@400;700&display=swap" media="print" onload="this.media='all'">
    
  

  
  
  
  
  
  <link rel="stylesheet" href="/css/wowchemy.9a66e344a68eb664b392d406a3f80726.css" />

  



  

  

  




  
  
  

  

  
    <link rel="manifest" href="/index.webmanifest" />
  

  <link rel="icon" type="image/png" href="/media/icon_hua2ec155b4296a9c9791d015323e16eb5_11927_32x32_fill_lanczos_center_2.png" />
  <link rel="apple-touch-icon" type="image/png" href="/media/icon_hua2ec155b4296a9c9791d015323e16eb5_11927_180x180_fill_lanczos_center_2.png" />

  <link rel="canonical" href="/post/blupadc/" />

  
  
  
  
  
  
  
  
    
    
  
  
  <meta property="twitter:card" content="summary" />
  
  <meta property="og:site_name" content="QSmei-Academic" />
  <meta property="og:url" content="/post/blupadc/" />
  <meta property="og:title" content="R package:blupADC-Feature 1 | QSmei-Academic" />
  <meta property="og:description" content="Convert genotype data format" /><meta property="og:image" content="/media/icon_hua2ec155b4296a9c9791d015323e16eb5_11927_512x512_fill_lanczos_center_2.png" />
    <meta property="twitter:image" content="/media/icon_hua2ec155b4296a9c9791d015323e16eb5_11927_512x512_fill_lanczos_center_2.png" /><meta property="og:locale" content="en-us" />
  
    
      <meta
        property="article:published_time"
        content="2021-04-14T00:00:00&#43;00:00"
      />
    
    <meta property="article:modified_time" content="2021-04-14T14:59:41&#43;08:00">
  

  


    






  




<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "/post/blupadc/"
  },
  "headline": "R package:blupADC-Feature 1",
  
  "datePublished": "2021-04-14T00:00:00Z",
  "dateModified": "2021-04-14T14:59:41+08:00",
  
  "author": {
    "@type": "Person",
    "name": "Mei Quanshun"
  },
  
  "publisher": {
    "@type": "Organization",
    "name": "QSmei-Academic",
    "logo": {
      "@type": "ImageObject",
      "url": "/media/icon_hua2ec155b4296a9c9791d015323e16eb5_11927_192x192_fill_lanczos_center_2.png"
    }
  },
  "description": "Convert genotype data format"
}
</script>

  

  

  

  





  <title>R package:blupADC-Feature 1 | QSmei-Academic</title>
</head>


<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" class="page-wrapper   " data-wc-page-id="b107928fdee05958ff3b5aa1f0ccc19d" >

  
  
  
  
  
  
  
  
  <script src="/js/wowchemy-init.min.226a9011996d125bf3fe4a5f22353a49.js"></script>

  


<aside class="search-modal" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#" aria-label="Close"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search" class="form-control"
        aria-label="Search...">
        
      </div>

      
      

      

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>



  <div class="page-header">
    












<nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id="navbar-main">
  <div class="container-xl">

    
    <div class="d-none d-lg-inline-flex">
      <a class="navbar-brand" href="/">QSmei-Academic</a>
    </div>
    

    
    <button type="button" class="navbar-toggler" data-toggle="collapse"
            data-target="#navbar-content" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
    <span><i class="fas fa-bars"></i></span>
    </button>
    

    
    <div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none">
      <a class="navbar-brand" href="/">QSmei-Academic</a>
    </div>
    

    
    
    <div class="navbar-collapse main-menu-item collapse justify-content-start" id="navbar-content">

      
      <ul class="navbar-nav d-md-inline-flex">
        

        

        
        
        
          
        

        

        
        
        
        

        
          
            
            
          
          
            
            
            
              
            
            
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#posts"><span>Posts</span></a>
        </li>

        
        

        

        
        
        
          
        

        

        
        
        
        

        
          
            
            
          
          
            
            
            
              
            
            
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#projects"><span>Projects</span></a>
        </li>

        
        

        

        
        
        
          
        

        

        
        
        
        

        
          
            
            
          
          
            
            
            
              
            
            
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#talks"><span>Talks</span></a>
        </li>

        
        

        

        
        
        
          
        

        

        
        
        
        

        
          
            
            
          
          
            
            
            
              
            
            
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#publications"><span>Publications</span></a>
        </li>

        
        

        

        
        
        
          
        

        

        
        
        
        

        
          
            
            
          
          
            
            
            
              
            
            
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#about"><span>About Me</span></a>
        </li>

        
        

      

        
      </ul>
    </div>

    <ul class="nav-icons navbar-nav flex-row ml-auto d-flex pl-md-2">

      
      
        
          
          <li class="nav-item d-none d-lg-inline-flex">
            <a class="nav-link" href="https://twitter.com/GeorgeCushen" data-toggle="tooltip" data-placement="bottom" title="Follow me on Twitter" target="_blank" rel="noopener" aria-label="Follow me on Twitter">
              <i class="fab fa-twitter" aria-hidden="true"></i>
            </a>
          </li>
        
      

      
      
      <li class="nav-item">
        <a class="nav-link js-search" href="#" aria-label="Search"><i class="fas fa-search" aria-hidden="true"></i></a>
      </li>
      

      
      
      <li class="nav-item dropdown theme-dropdown">
        <a href="#" class="nav-link" data-toggle="dropdown" aria-haspopup="true" aria-label="Display preferences">
          <i class="fas fa-moon" aria-hidden="true"></i>
        </a>
        <div class="dropdown-menu">
          <a href="#" class="dropdown-item js-set-theme-light">
            <span>Light</span>
          </a>
          <a href="#" class="dropdown-item js-set-theme-dark">
            <span>Dark</span>
          </a>
          <a href="#" class="dropdown-item js-set-theme-auto">
            <span>Automatic</span>
          </a>
        </div>
      </li>
      

      
      

    </ul>

  </div>
</nav>


  </div>

  <div class="page-body">
    <article class="article">

  












  

  
  
  
<div class="article-container pt-3">
  <h1>R package:blupADC-Feature 1</h1>

  

  
    


<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
          Last updated on
      
    
    Apr 14, 2021
  </span>
  

  

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    10 min read
  </span>
  

  
  
  
  
  
  

  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fas fa-folder mr-1"></i><a href="/category/data-format/">Data format</a></span>
  

</div>

    





  
</div>



  <div class="article-container">

    <div class="article-style">
      
<script src="/post/blupadc/index_files/header-attrs/header-attrs.js"></script>

<div id="TOC">
<ul>
<li><a href="#overview">Overview</a></li>
<li><a href="#example">Example</a></li>
<li><a href="#output">Output</a></li>
<li><a href="#parameter">Parameter</a>
<ul>
<li><a href="#basic">🤡Basic</a></li>
<li><a href="#advanced">💨Advanced</a></li>
</ul></li>
</ul>
</div>

<div id="overview" class="section level2">
<h2>Overview</h2>
<p><code>genotype_data_format_conversion</code> is the basic function of package:<code>blupADC</code>. By applying <code>genotype_data_format_conversion</code> , we can convert multiple genotype data formats in an easy way, including Hapmap, Plink, BLUPF90, Numeric and VCF.</p>
<p><strong>👉 Note: For analyzing VCF format data, <code>blupADC</code> has encapsulated software Plink <code>version-1.9</code> and VCFtools <code>version-0.1.17</code>. If you want change this version, you could set the related parameters in the part of Advanced parameter.</strong></p>
</div>
<div id="example" class="section level2">
<h2>Example</h2>
<pre class="r"><code>library(blupADC)
sum_data=genotype_data_format_conversion(
         input_data_hmp=data_hmp,  #provided hapmap data object 
         output_data_type=c(&quot;Plink&quot;,&quot;BLUPF90&quot;,&quot;Numeric&quot;),# output data format
         return_result = TRUE,      # return result 
         cpu_cores=1                # number of cpu 
         )</code></pre>
<pre><code>## Start numeric(0,1,2) format conversion...... 
## Complete BLUPF90  format conversion 
## Complete Plink Data Format Conversion</code></pre>
</div>
<div id="output" class="section level2">
<h2>Output</h2>
<pre class="r"><code>str(sum_data) 
## List of 5
##  $ hmp    :&#39;data.frame&#39;: 6 obs. of  311 variables:
##   ..$ rs#      : chr [1:6] &quot;SNP1&quot; &quot;SNP2&quot; &quot;SNP3&quot; &quot;SNP4&quot; ...
##   ..$ alleles  : logi [1:6] NA NA NA NA NA NA
##   ..$ chrom    : int [1:6] 1 1 1 1 1 1
##   ..$ pos      : int [1:6] 224488 293696 333333 464830 722623 838596
##   ..$ strand   : logi [1:6] NA NA NA NA NA NA
##   ..$ assembly : logi [1:6] NA NA NA NA NA NA
##   ..$ center   : logi [1:6] NA NA NA NA NA NA
##   ..$ protLSID : logi [1:6] NA NA NA NA NA NA
##   ..$ assayLSID: logi [1:6] NA NA NA NA NA NA
##   ..$ panelLSID: logi [1:6] NA NA NA NA NA NA
##   ..$ QCcode   : logi [1:6] NA NA NA NA NA NA
##   ..$ 3098     : chr [1:6] &quot;CC&quot; &quot;GG&quot; &quot;GG&quot; &quot;CC&quot; ...
##   ..$ 3498     : chr [1:6] &quot;AC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 3297     : chr [1:6] &quot;AC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 2452     : chr [1:6] &quot;CC&quot; &quot;GG&quot; &quot;GG&quot; &quot;CC&quot; ...
##   ..$ 4255     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 2946     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 3926     : chr [1:6] &quot;AC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 3426     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 4213     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 2895     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 4172     : chr [1:6] &quot;CC&quot; &quot;TT&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 3620     : chr [1:6] &quot;AA&quot; &quot;GG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 4797     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 4660     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 4010     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 4095     : chr [1:6] &quot;CC&quot; &quot;TT&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3029     : chr [1:6] &quot;AC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 3930     : chr [1:6] &quot;CC&quot; &quot;TT&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 4404     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 2020     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 2455     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3747     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 1590     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 2707     : chr [1:6] &quot;AC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 2320     : chr [1:6] &quot;CC&quot; &quot;GG&quot; &quot;GG&quot; &quot;CC&quot; ...
##   ..$ 3397     : chr [1:6] &quot;CC&quot; &quot;GG&quot; &quot;GG&quot; &quot;CC&quot; ...
##   ..$ 3876     : chr [1:6] &quot;AC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 3658     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 2310     : chr [1:6] &quot;CC&quot; &quot;TT&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 1996     : chr [1:6] &quot;CC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3000     : chr [1:6] &quot;CC&quot; &quot;GG&quot; &quot;GG&quot; &quot;CC&quot; ...
##   ..$ 2485     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 4412     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3395     : chr [1:6] &quot;CC&quot; &quot;TT&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 4449     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3302     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3895     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 4328     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3892     : chr [1:6] &quot;AC&quot; &quot;TG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3764     : chr [1:6] &quot;AC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 3922     : chr [1:6] &quot;CC&quot; &quot;TT&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 4459     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 4236     : chr [1:6] &quot;AC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 3535     : chr [1:6] &quot;CC&quot; &quot;GG&quot; &quot;GG&quot; &quot;CC&quot; ...
##   ..$ 2957     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 2927     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3499     : chr [1:6] &quot;CC&quot; &quot;TT&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 4537     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 4288     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 4048     : chr [1:6] &quot;CC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3371     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3901     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 3344     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 4745     : chr [1:6] &quot;AA&quot; &quot;GG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 3118     : chr [1:6] &quot;AC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 3093     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3410     : chr [1:6] &quot;CC&quot; &quot;GG&quot; &quot;GG&quot; &quot;CC&quot; ...
##   ..$ 4024     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 3053     : chr [1:6] &quot;AC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 3571     : chr [1:6] &quot;CC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3740     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 4374     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 4550     : chr [1:6] &quot;CC&quot; &quot;GG&quot; &quot;GG&quot; &quot;CC&quot; ...
##   ..$ 3060     : chr [1:6] &quot;CC&quot; &quot;GG&quot; &quot;GG&quot; &quot;CC&quot; ...
##   ..$ 3460     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;GG&quot; &quot;CC&quot; ...
##   ..$ 3791     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3116     : chr [1:6] &quot;AA&quot; &quot;GG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 3078     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 1974     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 1990     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3433     : chr [1:6] &quot;AA&quot; &quot;GG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 4349     : chr [1:6] &quot;AC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 3340     : chr [1:6] &quot;AC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 2681     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 2628     : chr [1:6] &quot;CC&quot; &quot;TT&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 4813     : chr [1:6] &quot;CC&quot; &quot;GG&quot; &quot;GG&quot; &quot;CC&quot; ...
##   ..$ 2221     : chr [1:6] &quot;CC&quot; &quot;TT&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 2384     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 2721     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3463     : chr [1:6] &quot;AA&quot; &quot;GG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 2748     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 2973     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 1999     : chr [1:6] &quot;AC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 3474     : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 3689     : chr [1:6] &quot;AC&quot; &quot;TG&quot; &quot;TT&quot; &quot;CC&quot; ...
##   ..$ 1924     : chr [1:6] &quot;CC&quot; &quot;GG&quot; &quot;GG&quot; &quot;CC&quot; ...
##   ..$ 1948     : chr [1:6] &quot;AC&quot; &quot;GG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   ..$ 110      : chr [1:6] &quot;CC&quot; &quot;TG&quot; &quot;GT&quot; &quot;CC&quot; ...
##   .. [list output truncated]
##  $ ped    : chr [1:300, 1:18] &quot;3098&quot; &quot;3498&quot; &quot;3297&quot; &quot;2452&quot; ...
##  $ map    :&#39;data.frame&#39;: 6 obs. of  4 variables:
##   ..$ chrom : int [1:6] 1 1 1 1 1 1
##   ..$ rs#   : chr [1:6] &quot;SNP1&quot; &quot;SNP2&quot; &quot;SNP3&quot; &quot;SNP4&quot; ...
##   ..$ strand: num [1:6] 0.224 0.294 0.333 0.465 0.723 ...
##   ..$ pos   : int [1:6] 224488 293696 333333 464830 722623 838596
##  $ blupf90: chr [1:300, 1:2] &quot;3098                &quot; &quot;3498                &quot; &quot;3297                &quot; &quot;2452                &quot; ...
##   ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. ..$ : NULL
##   .. ..$ : chr [1:2] &quot;IND_name&quot; &quot;&quot;
##  $ numeric: num [1:300, 1:6] 2 1 1 2 1 2 1 1 2 2 ...
##   ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. ..$ : chr [1:300] &quot;3098&quot; &quot;3498&quot; &quot;3297&quot; &quot;2452&quot; ...
##   .. ..$ : chr [1:6] &quot;SNP1&quot; &quot;SNP2&quot; &quot;SNP3&quot; &quot;SNP4&quot; ...</code></pre>
<p>According to the result of output, we find that the output contains 5 parts, including:</p>
<ul>
<li><strong>hmp </strong> : <code>Hapmap</code> format genotype data</li>
</ul>
<p><code>The first column stands for the name of SNP, the thrid column stands for chromosome, the fourth column stands for the physical postion, and the twelth column and the after columns stand for the genotype data</code></p>
<table style="width:100%;">
<colgroup>
<col width="4%" />
<col width="7%" />
<col width="5%" />
<col width="6%" />
<col width="6%" />
<col width="8%" />
<col width="6%" />
<col width="8%" />
<col width="9%" />
<col width="9%" />
<col width="6%" />
<col width="4%" />
<col width="4%" />
<col width="4%" />
<col width="4%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">rs#</th>
<th align="left">alleles</th>
<th align="right">chrom</th>
<th align="right">pos</th>
<th align="left">strand</th>
<th align="left">assembly</th>
<th align="left">center</th>
<th align="left">protLSID</th>
<th align="left">assayLSID</th>
<th align="left">panelLSID</th>
<th align="left">QCcode</th>
<th align="left">3098</th>
<th align="left">3498</th>
<th align="left">3297</th>
<th align="left">2452</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">SNP1</td>
<td align="left">NA</td>
<td align="right">1</td>
<td align="right">224488</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">CC</td>
<td align="left">AC</td>
<td align="left">AC</td>
<td align="left">CC</td>
</tr>
<tr class="even">
<td align="left">SNP2</td>
<td align="left">NA</td>
<td align="right">1</td>
<td align="right">293696</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">GG</td>
<td align="left">TG</td>
<td align="left">TG</td>
<td align="left">GG</td>
</tr>
<tr class="odd">
<td align="left">SNP3</td>
<td align="left">NA</td>
<td align="right">1</td>
<td align="right">333333</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">GG</td>
<td align="left">TT</td>
<td align="left">TT</td>
<td align="left">GG</td>
</tr>
<tr class="even">
<td align="left">SNP4</td>
<td align="left">NA</td>
<td align="right">1</td>
<td align="right">464830</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">CC</td>
<td align="left">CC</td>
<td align="left">CC</td>
<td align="left">CC</td>
</tr>
<tr class="odd">
<td align="left">SNP5</td>
<td align="left">NA</td>
<td align="right">1</td>
<td align="right">722623</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">AA</td>
<td align="left">GG</td>
<td align="left">GG</td>
<td align="left">AA</td>
</tr>
<tr class="even">
<td align="left">SNP6</td>
<td align="left">NA</td>
<td align="right">1</td>
<td align="right">838596</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">CC</td>
<td align="left">TC</td>
<td align="left">TT</td>
<td align="left">CC</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>ped</strong> : <code>Plink</code> 格式的ped数据</li>
</ul>
<p><code>The first column stands for family name,the second column stands for the individual name,the seventh column and the after columns stand for the genotype data</code></p>
<table>
<tbody>
<tr class="odd">
<td align="left">3098</td>
<td align="left">3098</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">C</td>
<td align="left">C</td>
<td align="left">G</td>
<td align="left">G</td>
</tr>
<tr class="even">
<td align="left">3498</td>
<td align="left">3498</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">A</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="left">G</td>
</tr>
<tr class="odd">
<td align="left">3297</td>
<td align="left">3297</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">A</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="left">G</td>
</tr>
<tr class="even">
<td align="left">2452</td>
<td align="left">2452</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">C</td>
<td align="left">C</td>
<td align="left">G</td>
<td align="left">G</td>
</tr>
<tr class="odd">
<td align="left">4255</td>
<td align="left">4255</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">A</td>
<td align="left">C</td>
<td align="left">G</td>
<td align="left">G</td>
</tr>
<tr class="even">
<td align="left">2946</td>
<td align="left">2946</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">C</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="left">G</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>map </strong>: <code>Plink</code>格式的map数据</li>
</ul>
<p><code>The first column stands for chromosome, the second column stands for the name of SNP, the thrid column stands for the genetic positon(CM), and the fourth column stands for the physical position</code></p>
<table>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="left">SNP1</td>
<td align="right">0.224488</td>
<td align="right">224488</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">SNP2</td>
<td align="right">0.293696</td>
<td align="right">293696</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">SNP3</td>
<td align="right">0.333333</td>
<td align="right">333333</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">SNP4</td>
<td align="right">0.464830</td>
<td align="right">464830</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">SNP5</td>
<td align="right">0.722623</td>
<td align="right">722623</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">SNP6</td>
<td align="right">0.838596</td>
<td align="right">838596</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>blupf90 </strong>: <code>BLUPF90</code>格式的基因型数据</li>
</ul>
<p><code>The first column stands for individual name, the second column stands for the genotype data(numeric)</code></p>
<table>
<tbody>
<tr class="odd">
<td align="left">3098</td>
<td align="left">200000</td>
</tr>
<tr class="even">
<td align="left">3498</td>
<td align="left">112021</td>
</tr>
<tr class="odd">
<td align="left">3297</td>
<td align="left">112022</td>
</tr>
<tr class="even">
<td align="left">2452</td>
<td align="left">200000</td>
</tr>
<tr class="odd">
<td align="left">4255</td>
<td align="left">102011</td>
</tr>
<tr class="even">
<td align="left">2946</td>
<td align="left">212000</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>numeric </strong>: <code>Numeric</code> 格式的基因型数据</li>
</ul>
<p><code>rownames stands for the individual name, colnames stands for the name of SNP, 0,1,2 stand for the numeric genotype</code></p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">SNP1</th>
<th align="right">SNP2</th>
<th align="right">SNP3</th>
<th align="right">SNP4</th>
<th align="right">SNP5</th>
<th align="right">SNP6</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">3098</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">3498</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">3297</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">2452</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">4255</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">2946</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
</div>
<div id="parameter" class="section level2">
<h2>Parameter</h2>
<div id="basic" class="section level3">
<h3>🤡Basic</h3>
<ul>
<li><strong>1: input_data_plink_ped</strong></li>
</ul>
<p>User-provided <code>Plink</code>-ped format genotype data ，<code>data.frame</code> class.</p>
<ul>
<li><strong>2:input_data_plink_map</strong></li>
</ul>
<p>User-provided <code>Plink</code>-map format genotype data ，<code>data.frame</code> class.</p>
<ul>
<li><strong>3:input_data_hmp</strong></li>
</ul>
<p>User-provided <code>Hapmap</code> format genotype data ，<code>data.frame</code> class.</p>
<ul>
<li><strong>4:input_data_BLUPF90</strong></li>
</ul>
<p>User-provided <code>BLUPF90</code> format genotype data ，<code>data.frame</code> class.</p>
<ul>
<li><strong>5:input_data_numeric</strong></li>
</ul>
<p>User-provided <code>Numeric</code> format genotype data ，<code>matrix(numeric)</code> class.</p>
<p><strong>Note:</strong> <code>input_data_numeric</code> should contain both rownames and colnames.</p>
<p><strong>In addition, for convenience, users can provide the file name, file path, and file type of genotype data directly without reading them in R environment.</strong></p>
<ul>
<li><p><strong>6:input_data_type</strong></p>
<p>File type of provided genotype data ，<code>character</code> class.</p>
<ul>
<li>Hapmap</li>
<li>Plink</li>
<li>BLUPF90</li>
<li>Numeric</li>
<li>VCF</li>
</ul></li>
<li><p><strong>7:input_data_path</strong></p></li>
</ul>
<p>File path of provided genotype data ，<code>character</code> class.</p>
<ul>
<li><strong>8:input_data_name</strong></li>
</ul>
<p>File name of provided genotype data ，<code>character</code> class.</p>
<p><strong>Note:</strong> if input_data_type is Plink or VCF, user don’t need to include suffix in the file name of genotype data.</p>
<p>eg. for Plink type data, files name are test1.map and test1.ped，we should set<code>input_data_name="test1"</code>.</p>
<ul>
<li><strong>9:return_result</strong></li>
</ul>
<p>Whether return result, <code>logical</code> class. Default is FALSE.</p>
<p>Additionally, for convenience, users can save output genotype data into local computer .</p>
<ul>
<li><strong>10:output_data_path</strong></li>
</ul>
<p>File path of output genotype data, <code>character</code> class.</p>
<ul>
<li><strong>11:output_data_name</strong></li>
</ul>
<p>File name of output genotype data, <code>character</code> class.</p>
<ul>
<li><p><strong>12:output_data_type</strong></p>
<p>File type of output genotype data, <code>character</code> class.</p>
<ul>
<li>Hapmap</li>
<li>Plink</li>
<li>BLUPF90</li>
<li>Numeric</li>
<li>VCF</li>
</ul></li>
</ul>
<p><strong>Note</strong>: users can output multiple formats of genotype data simultaneously. e.g.<code>output_data_type=c("Hapmap","Plink","BLUPF90","Numeric")</code>, outout 4 types of genotype data simultaneously .</p>
</div>
<div id="advanced" class="section level3">
<h3>💨Advanced</h3>
<ul>
<li><strong>13:cpu_cores</strong></li>
</ul>
<p>Number of cpu in calculating, <code>numeric</code> class. Default is 1.</p>
<ul>
<li><strong>14:miss_base</strong></li>
</ul>
<p>Missing genotype character, <code>character</code> class. Default is “NN”.</p>
<ul>
<li><strong>15:miss_base_num</strong></li>
</ul>
<p>Missing genotype number after numeric conversion, <code>numeric</code> class. Default is 5.</p>
<ul>
<li><strong>16:plink_software_path</strong></li>
</ul>
<p>Path of plink software in user computer. <code>character</code> class.</p>
<ul>
<li><strong>17:plink_software_name</strong></li>
</ul>
<p>Name of plink software in user computer. <code>character</code> class.</p>
<ul>
<li><strong>18:vcftools_software_path</strong></li>
</ul>
<p>Path of vcftools software in user computer. <code>character</code> class.</p>
<ul>
<li><strong>19:vcftools_software_name</strong></li>
</ul>
<p>Name of vcftools software in user computer. <code>character</code> class.</p>
</div>
</div>

    </div>

    








<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=/post/blupadc/&amp;text=R%20package:blupADC-Feature%201" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=/post/blupadc/&amp;t=R%20package:blupADC-Feature%201" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=R%20package:blupADC-Feature%201&amp;body=/post/blupadc/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=/post/blupadc/&amp;title=R%20package:blupADC-Feature%201" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="whatsapp://send?text=R%20package:blupADC-Feature%201%20/post/blupadc/" target="_blank" rel="noopener" class="share-btn-whatsapp">
          <i class="fab fa-whatsapp"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://service.weibo.com/share/share.php?url=/post/blupadc/&amp;title=R%20package:blupADC-Feature%201" target="_blank" rel="noopener" class="share-btn-weibo">
          <i class="fab fa-weibo"></i>
        </a>
      </li>
    
  </ul>
</div>











  
  



  
  
  
    
  
  
  
  <div class="media author-card content-widget-hr">
    
      
      <img class="avatar mr-3 avatar-circle" src="/author/mei-quanshun/avatar_hu5f20744e08524ff7ac6fce3b60ea5d69_3615750_270x270_fill_q75_lanczos_center.JPG" alt="Mei Quanshun">
    

    <div class="media-body">
      <h5 class="card-title">Mei Quanshun</h5>
      <h6 class="card-subtitle">PhD student in animal breeding</h6>
      <p class="card-text">My research interests include genomic selection and machine learning in animal breeding.</p>
      <ul class="network-icon" aria-hidden="true">
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="/#contact" >
        <i class="fas fa-envelope"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://twitter.com/GeorgeCushen" target="_blank" rel="noopener">
        <i class="fab fa-twitter"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://scholar.google.co.uk/citations?user=sIwtMXoAAAAJ" target="_blank" rel="noopener">
        <i class="fas fa-graduation-cap"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://github.com/gcushen" target="_blank" rel="noopener">
        <i class="fab fa-github"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://www.linkedin.com/" target="_blank" rel="noopener">
        <i class="fab fa-linkedin"></i>
      </a>
    </li>
  
</ul>

    </div>
  </div>
















  
  





  </div>
</article>
  </div>

  <div class="page-footer">
    
    
    <div class="container">
      <footer class="site-footer">

  

  

  <p class="powered-by">
    
  </p>

  
  






  <p class="powered-by">
    
    
    
      
      
      
      
      
      
      Published with <a href="https://wowchemy.com/?utm_campaign=poweredby" target="_blank" rel="noopener">Wowchemy</a> — the free, <a href="https://github.com/wowchemy/wowchemy-hugo-modules" target="_blank" rel="noopener">open source</a> website builder that empowers creators.
    
  </p>
</footer>

    </div>
    
  </div>

  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

      

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha256-9/aliU8dGd2tb6OSsuzixeV4y/faTqgFtohetphbbj0=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/instant.page/5.1.0/instantpage.min.js" integrity="sha512-1+qUtKoh9XZW7j+6LhRMAyOrgSQKenQ4mluTR+cvxXjP1Z54RxZuzstR/H9kgPXQsVB8IW7DMDFUJpzLjvhGSQ==" crossorigin="anonymous"></script>

      
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js" integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin="anonymous"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js" integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin="anonymous"></script>
      

      
      

      

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/highlight.min.js" integrity="sha512-TDKKr+IvoqZnPzc3l35hdjpHD0m+b2EC2SrLEgKDRWpxf2rFCxemkgvJ5kfU48ip+Y+m2XVKyOCD85ybtlZDmw==" crossorigin="anonymous"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/languages/r.min.js"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/languages/python.min.js"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/languages/latex.min.js"></script>
        
      

    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.7.1/leaflet.min.js" integrity="sha512-SeiQaaDh73yrb56sTW/RgVdi/mMqNeM2oBwubFHagc5BkixSpP1fvqF47mKzPGWYSSy4RwbBunrJBQ4Co8fRWA==" crossorigin="anonymous"></script>
    

    
    

    
    
    
      
      <script id="search-hit-fuse-template" type="text/x-template">
        <div class="search-hit" id="summary-{{key}}">
          <div class="search-hit-content">
            <div class="search-hit-name">
              <a href="{{relpermalink}}">{{title}}</a>
              <div class="article-metadata search-hit-type">{{type}}</div>
              <p class="search-hit-description">{{snippet}}</p>
            </div>
          </div>
        </div>
      </script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
      
    

    
    

    
    
    
    

    
    <script src="/js/bootstrap.bundle.min.6aed84840afc03ab4d5750157f69c120.js"></script>

    
    
      
      
      
      
      
      
      
    

    
    
    
    
    
    
    
    
      
      
    
    
    <script src="/en/js/wowchemy.min.b61a8f62b6e5c0cd322c8158c5b5dfb6.js"></script>

    
  <script async defer src="https://buttons.github.io/buttons.js"></script>




</body>
</html>
