---
title: 'R package:blupADC-ä½¿ç”¨è¯´æ˜5'
author: qsmei
date: '2021-04-18'
slug: relationship_matrix
categories:
  - Relationship matrix
tags: []
subtitle: ''
summary: 'è®¡ç®—äº²ç¼˜å…³ç³»çŸ©é˜µ(A,G and H)'
authors: []
lastmod: '2021-04-18T16:36:39+08:00'
featured: no
image:
  caption: ''
  focal_point: ''
  preview_only: no
projects: []
---

<script src="/rmarkdown-libs/header-attrs/header-attrs.js"></script>


<p>ğŸ™‰åœ¨åŠ¨æ¤ç‰©è‚²ç§ä¸­ï¼Œäº²ç¼˜å…³ç³»çŸ©é˜µçš„æ„å»ºæ˜¯å…¶ä¸­çš„å…³é”®æ­¥éª¤ã€‚åœ¨æœ¬ç« ï¼Œæˆ‘ä»¬å°†ä¸»è¦ä»‹ç»å¦‚ä½•åˆ©ç”¨BLUP_ADCä¸­çš„cal_kinshipå‡½æ•°å®Œæˆå„ç§äº²ç¼˜å…³ç³»çŸ©é˜µçš„æ„å»ºï¼ŒåŒ…æ‹¬ï¼šåŠ æ€§äº²ç¼˜å…³ç³»çŸ©é˜µ(ç³»è°±ï¼ŒåŸºå› ç»„ï¼Œä¸€æ­¥æ³•)åŠæ˜¾æ€§äº²ç¼˜å…³ç³»çŸ©é˜µ(ç³»è°±ï¼ŒåŸºå› ç»„ï¼Œä¸€æ­¥æ³•)ç­‰ã€‚æ­¤å¤–ï¼Œcal_kinshipå‡½æ•°è¿˜èƒ½æ–¹ä¾¿çš„è®¡ç®—å„ç§ç±»å‹çš„è¿‘äº¤ç³»æ•°ï¼ŒåŒ…æ‹¬ï¼šç³»è°±è¿‘äº¤ç³»æ•°ï¼ŒåŸºå› ç»„è¿‘äº¤ç³»æ•°(Homozygous,Digonal)åŠä¸€æ­¥æ³•è¿‘äº¤ç³»æ•°(Digonal)ã€‚</p>
<p>ğŸ˜¾é¦–å…ˆï¼Œæˆ‘ä»¬è¿˜æ˜¯é€šè¿‡ä¸€ä¸ªå°ä¾‹å­æ¥çœ‹è¯¥å‡½æ•°çš„ç”¨æ³•:</p>
<pre class="r"><code>source(&quot;D:/test_code/BLUP_ADC_code/data_relationship_matrix.txt&quot;)
source(&quot;D:/test_code/BLUP_ADC_code/data_format_conversion.txt&quot;)
Rcpp::sourceCpp(&quot;D:/test_code/BLUP_ADC_code/code/common.function.cpp&quot;)
data_hmp=data.table::fread(&quot;D:/test_code/BLUP_ADC_code/test_data/data_hmp.txt&quot;,data.table=F)
data_result=cal_kinship(input_data_hmp=data_hmp,
                        kinship_type=c(&quot;G_A&quot;),
                        inbred_type=c(&quot;Homozygous&quot;),
                        return_result=TRUE)
## 
## è½½å…¥ç¨‹è¾‘åŒ…ï¼š&#39;dplyr&#39;
## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag
## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union
## Start numeric(0,1,2) format conversion...... 
## Constructing genomic additive relationship matrix......
## Start calculate genomic inbreeding coefficients by homozygous site
str(data_result)
## List of 8
##  $ G_A   :List of 2
##   ..$ A   : num [1:500, 1:500] 1.0783 -0.0685 -0.0336 -0.0836 0.0455 ...
##   ..$ Ainv: NULL
##  $ G_Domi:List of 2
##   ..$ Domi    : NULL
##   ..$ Domi_inv: NULL
##  $ G_Devi:List of 2
##   ..$ Devi    : NULL
##   ..$ Devi_inv: NULL
##  $ P_A   :List of 2
##   ..$ A   : NULL
##   ..$ Ainv: NULL
##  $ P_D   :List of 2
##   ..$ D   : NULL
##   ..$ Dinv: NULL
##  $ H_A   :List of 2
##   ..$ A   : NULL
##   ..$ Ainv: NULL
##  $ H_D   :List of 2
##   ..$ D   : NULL
##   ..$ Dinv: NULL
##  $ Inbred:List of 4
##   ..$ Homozygous:&#39;data.frame&#39;:   500 obs. of  2 variables:
##   .. ..$ Id         : Factor w/ 500 levels &quot;Ind1&quot;,&quot;Ind10&quot;,..: 1 112 223 334 445 457 468 479 490 2 ...
##   .. ..$ Homo_inbred: num [1:500] 0.649 0.629 0.597 0.66 0.657 ...
##   ..$ G_diag    : NULL
##   ..$ Pedigree  : NULL
##   ..$ H_diag    : NULL</code></pre>
<p>åœ¨æ„å»ºåŸºå› ç»„äº²ç¼˜å…³ç³»çŸ©é˜µåŠä¸€æ­¥æ³•äº²ç¼˜å…³ç³»çŸ©é˜µçš„æ—¶å€™ï¼Œæˆ‘ä»¬å¿…é¡»è¦äº‹å…ˆæä¾›åŸºå› å‹æ•°æ®ï¼Œè¿™éƒ¨åˆ†çš„å‚æ•°ä¸genotype_data_format_conversionå‡½æ•°ä¸­çš„å‚æ•°ç”¨æ³•ä¸€è‡´ï¼Œè¯¦ç»†äº†è§£å¤§å®¶å¯å‚é˜…ï¼Ÿï¼Ÿï¼Ÿã€‚</p>
<p>å®Œæˆäº†åŸºå› å‹æ•°æ®çš„æä¾›åï¼Œæˆ‘ä»¬å¯ä»¥é€šè¿‡ä»¥ä¸‹å‡ ä¸ªå‚æ•°æ¥æŒ‡å®šæ„å»ºäº²ç¼˜å…³ç³»çŸ©é˜µç±»å‹åŠè¿‘äº¤ç³»æ•°çš„ç±»å‹ï¼Œå¦‚ï¼š</p>
<ul>
<li><strong>å‚æ•°1ï¼škinship_type</strong></li>
</ul>
<p>æŒ‡å®šæ„å»ºäº²ç¼˜å…³ç³»çŸ©é˜µçš„ç±»å‹ï¼Œå­—ç¬¦å‘é‡ï¼Œå¯é€‰ç±»å‹åŒ…æ‹¬ï¼š</p>
<ol style="list-style-type: decimal">
<li><p>â€œG_Aâ€ :åŸºå› ç»„åŠ æ€§äº²ç¼˜å…³ç³»çŸ©é˜µ</p></li>
<li><p>â€œG_Domiâ€ :åŸºå› ç»„æ˜¾æ€§äº²ç¼˜å…³ç³»çŸ©é˜µ(æŒ‰ç…§0,1,0æ–¹å¼ç¼–ç )</p></li>
<li><p>â€œG_Deviâ€ :åŸºå› ç»„æ˜¾æ€§ç¦»å·®äº²ç¼˜å…³ç³»çŸ©é˜µ(æŒ‰ç…§-2q<sup>2</sup>,2pq,-2p<sup>2</sup>,æ–¹å¼ç¼–ç )</p></li>
<li><p>â€œH_Aâ€ :ä¸€æ­¥æ³•åŠ æ€§äº²ç¼˜å…³ç³»çŸ©é˜µ</p></li>
<li><p>â€œH_Dâ€ :ä¸€æ­¥æ³•æ˜¾æ€§äº²ç¼˜å…³ç³»çŸ©é˜µ</p></li>
<li><p>â€œP_Aâ€ :ç³»è°±åŠ æ€§äº²ç¼˜å…³ç³»çŸ©é˜µ</p></li>
<li><p>â€œP_Dâ€ :ç³»è°±æ˜¾æ€§äº²ç¼˜å…³ç³»çŸ©é˜µ</p></li>
</ol>
<p>Note:å¦‚æœè®¡ç®—ç³»è°±åŠä¸€æ­¥æ³•äº²ç¼˜å…³ç³»çŸ©é˜µï¼Œå¿…é¡»è¦æä¾›ç³»è°±æ•°æ®ã€‚å…³äºå¦‚ä½•æä¾›ç³»è°±æ•°æ®ï¼Œæˆ‘ä»¬ä¼šåœ¨åé¢è¿›è¡Œè®²è§£ã€‚</p>
<ul>
<li><strong>å‚æ•°2ï¼škinship_inverse</strong></li>
</ul>
<p>æ˜¯å¦è®¡ç®—äº²ç¼˜å…³ç³»çŸ©é˜µçš„é€†çŸ©é˜µï¼Œé€»è¾‘å€¼ï¼Œé»˜è®¤ä¸ºFALSE</p>
<ul>
<li><strong>å‚æ•°3ï¼šinbred_type</strong></li>
</ul>
<p>æŒ‡å®šè®¡ç®—çš„è¿‘äº¤ç³»æ•°çš„ç±»å‹ï¼Œå­—ç¬¦å‘é‡ï¼Œå¯é€‰ç±»å‹åŒ…æ‹¬:</p>
<ol style="list-style-type: decimal">
<li><p>â€œHomozygousâ€ :æ ¹æ®çº¯åˆå­ä½ç‚¹è®¡ç®—</p></li>
<li><p>â€œG_Diagâ€ :GçŸ©é˜µå¯¹è§’çº¿-1</p></li>
<li><p>â€œH_diagâ€ :HçŸ©é˜µå¯¹è§’çº¿-1</p></li>
<li><p>â€œPedigreeâ€ :AçŸ©é˜µå¯¹è§’çº¿-1</p></li>
</ol>
<ul>
<li><strong>å‚æ•°5ï¼šinput_pedigree</strong></li>
</ul>
<p>ç”¨æˆ·æä¾›çš„ç³»è°±æ•°æ®ï¼Œå…·ä½“çš„ç³»è°±æ•°æ®ç±»å‹å¯ä»¥å‚é˜…ä¹‹å‰çš„ä»‹ç»ï¼šï¼Ÿï¼Ÿï¼Ÿ</p>
<ul>
<li><strong>å‚æ•°6ï¼špedigree_rename</strong></li>
</ul>
<p>æ˜¯å¦å¯¹ç³»è°±æ•°æ®è¿›è¡Œé‡å‘½å,é€»è¾‘å€¼ï¼Œé»˜è®¤ä¸ºTRUE(å¯¹ç³»è°±è¿›è¡Œé‡å‘½å)</p>
<ul>
<li><strong>å‚æ•°7ï¼šIND_geno_rename</strong></li>
</ul>
<p>æ ¹æ®ç³»è°±çš„é‡å‘½åç»“æœï¼Œå°†åŸºå› å‹æ•°æ®ä¸­çš„ä¸ªä½“å·ä¹Ÿè¿›è¡Œé‡å‘½åã€‚é€»è¾‘å€¼ï¼Œé»˜è®¤ä¸ºFALSE(ä¸è¿›è¡Œé‡å‘½å)</p>
<ul>
<li><strong>å‚æ•°8ï¼šrename_to_origin</strong></li>
</ul>
<p>å°†çŸ©é˜µä¸­é‡å‘½åçš„ä¸ªä½“å·è½¬æ¢å›åŸå§‹çš„ä¸ªä½“å·ã€‚é€»è¾‘å€¼ï¼Œé»˜è®¤ä¸ºFALSE(ä¸è¿›è¡Œè½¬æ¢)</p>
<ul>
<li><strong>å‚æ•°9ï¼šoutput_matrix_type</strong></li>
</ul>
<p>è¾“å‡ºçŸ©é˜µçš„æ ¼å¼ã€‚å­—ç¬¦ä¸²å‘é‡ï¼Œå¯é€‰å‚æ•°ä¸ºï¼š</p>
<ol style="list-style-type: decimal">
<li><p>â€œcol_allâ€ :æŒ‰ç…§n*nçš„æ ¼å¼è¾“å‡ºäº²ç¼˜å…³ç³»çŸ©é˜µ</p></li>
<li><p>â€œcol_threeâ€ : æŒ‰ç…§3åˆ—çŸ©é˜µçš„æ ¼å¼è¾“å‡ºäº²ç¼˜å…³ç³»çŸ©é˜µï¼Œç¬¬1åˆ—å’Œç¬¬2åˆ—ä¸ºä¸ªä½“å·ï¼Œç¬¬3åˆ—ä¸ºäº²ç¼˜ç³»æ•°ã€‚DMUå’ŒBLUPf90è½¯ä»¶å‡éœ€æä¾›è¿™ç§æ ¼å¼çš„äº²ç¼˜å…³ç³»çŸ©é˜µã€‚å½¢å¦‚ï¼š</p>
<table>
<thead>
<tr class="header">
<th>V1</th>
<th>V2</th>
<th>V3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1001</td>
<td>1001</td>
<td>0.989</td>
</tr>
<tr class="even">
<td>1001</td>
<td>1002</td>
<td>0.421</td>
</tr>
<tr class="odd">
<td>1001</td>
<td>1003</td>
<td>0.567</td>
</tr>
</tbody>
</table></li>
</ol>
<p>é»˜è®¤å‚æ•°ä¸º â€œcol_allâ€</p>
<ul>
<li><strong>å‚æ•°10ï¼šoutput_matrix_path</strong></li>
</ul>
<p>äº²ç¼˜å…³ç³»çŸ©é˜µä¿å­˜åˆ°æœ¬åœ°çš„è·¯å¾„, å­—ç¬¦ä¸²å‘é‡</p>
<ul>
<li><strong>å‚æ•°11ï¼šoutput_matrix_name</strong></li>
</ul>
<p>äº²ç¼˜å…³ç³»çŸ©é˜µä¿å­˜åˆ°æœ¬åœ°çš„åç§°ï¼Œå­—ç¬¦ä¸²å‘é‡</p>
<div id="å…¶ä»–å‚æ•°" class="section level3">
<h3>å…¶ä»–å‚æ•°</h3>
<ul>
<li><strong>å‚æ•°11ï¼šcpu_cores</strong></li>
</ul>
<p>è°ƒç”¨çš„cpuä¸ªæ•°ï¼Œæ•°å­—æ ¼å¼ï¼Œé»˜è®¤ä¸º1</p>
<ul>
<li><strong>å‚æ•°11ï¼škinship_base</strong></li>
</ul>
<p>æŒ‰ç…§åŸºç¡€ç¾¤çš„æ–¹å¼æ„å»ºåŸºå› ç»„äº²ç¼˜å…³ç³»çŸ©é˜µ(p=q=0.5)ï¼Œé€»è¾‘å€¼ï¼Œé»˜è®¤ä¸ºFALSE</p>
<ul>
<li><strong>å‚æ•°11ï¼škinship_trace</strong></li>
</ul>
<p>æŒ‰ç…§çŸ©é˜µçš„è¿¹å’Œçš„æ–¹å¼å¯¹åŸºå› ç»„äº²ç¼˜å…³ç³»çŸ©é˜µè¿›è¡Œæ ‡å‡†åŒ–ï¼Œé€»è¾‘å€¼ï¼Œé»˜è®¤ä¸ºFALse</p>
<ul>
<li><strong>å‚æ•°4ï¼škinship_metafounder</strong></li>
</ul>
<p>æ˜¯å¦æŒ‰ç…§metafounderçš„æ–¹æ³•è®¡ç®—ä¸€æ­¥æ³•äº²ç¼˜å…³ç³»çŸ©é˜µï¼Œé€»è¾‘å€¼ï¼Œé»˜è®¤ä¸ºFALSE</p>
<ul>
<li><strong>å‚æ•°11ï¼šw</strong></li>
</ul>
<p>æ„å»ºä¸€æ­¥æ³•äº²ç¼˜å…³ç³»çŸ©é˜µæ—¶ï¼Œè°ƒæ•´GçŸ©é˜µå’ŒAçŸ©é˜µçš„æ¯”ä¾‹ã€‚numericæ ¼å¼ï¼Œé»˜è®¤ä¸º0.05</p>
<ul>
<li><strong>å‚æ•°11ï¼šgene_dropping</strong></li>
</ul>
<p>ä½¿ç”¨ gene dropping çš„æ–¹æ³•æ„å»ºç³»è°±æ˜¾æ€§äº²ç¼˜å…³ç³»çŸ©é˜µã€‚é€»è¾‘å€¼ï¼Œé»˜è®¤å€¼ä¸ºFALSE</p>
<ul>
<li><strong>å‚æ•°11ï¼šgene_dropping_iteration</strong></li>
</ul>
<p>gene droppingæ–¹æ³•çš„è¿­ä»£æ¬¡æ•°ã€‚numericæ ¼å¼ï¼Œé»˜è®¤å€¼ä¸º1000</p>
</div>
