---
title: 'R package:blupADC-Feature 1'
author: qsmei
date: '2021-04-14'
slug: blupADC
categories: ["Data format"]
tags: []
subtitle: ''
summary: 'Genomic data format conversion'
authors: []
lastmod: '2021-04-14T14:59:41+08:00'
featured: no
image:
  caption: ''
  focal_point: ''
  preview_only: no
projects: []
output:
  blogdown::html_page:
    toc: true
---

<script src="{{< blogdown/postref >}}index_files/header-attrs/header-attrs.js"></script>

<div id="TOC">
<ul>
<li><a href="#overview">Overview</a></li>
<li><a href="#example">Example</a></li>
<li><a href="#output">Output</a></li>
<li><a href="#parameter">Parameter</a>
<ul>
<li><a href="#basic">🤡Basic</a></li>
<li><a href="#advanced">💨Advanced</a></li>
</ul></li>
</ul>
</div>

<div id="overview" class="section level2">
<h2>Overview</h2>
<p><code>geno_format</code> is the basic function of package:<code>blupADC</code>. By applying <code>geno_format</code> , we can convert multiple genotype data formats in an easy way, including Hapmap, Plink, BLUPF90, Numeric, Haplotype and VCF.</p>
</div>
<div id="example" class="section level2">
<h2>Example</h2>
<div id="format-conversion-based-on-provided-r-variable" class="section level4">
<h4>Format conversion based on provided R variable</h4>
<pre class="r"><code>library(blupADC)
format_result=geno_format(
         input_data_hmp=example_data_hmp,  #provided hapmap data object 
         output_data_type=c(&quot;Plink&quot;,&quot;BLUPF90&quot;,&quot;Numeric&quot;,&quot;VCF&quot;),# output data format
         return_result = TRUE,      # return result 
         cpu_cores=1                # number of cpu 
         )</code></pre>
</div>
<div id="format-conversion-based-on-provided-data-path-and-data-name" class="section level4">
<h4>Format conversion based on provided data path and data name</h4>
<pre class="r"><code>#convert phased VCF data to haplotype format and  haplotype-based numeric format
library(blupADC)
data_path=system.file(&quot;extdata&quot;, package = &quot;blupADC&quot;)  #  path of example files 
phased_result=geno_format(
         input_data_path=data_path,      # input data path 
         input_data_name=&quot;example.vcf&quot;,  # input data name,for vcf data 
         input_data_type=&quot;VCF&quot;,          # input data type
         phased_genotype=TRUE,           # whether the vcf data has been phased
         haplotype_window_nSNP=5,        # according to nSNP define block,
         output_data_type=c(&quot;Haplotype&quot;,&quot;Numeric&quot;),# output data format
         return_result=TRUE,             #save result as a R environment variable
         cpu_cores=1                     # number of cpu 
                  )</code></pre>
</div>
<div id="format-conversion-via-bigmemory-method" class="section level4">
<h4>Format conversion via bigmemory method</h4>
<pre class="r"><code>library(blupADC)
data_path=system.file(&quot;extdata&quot;, package = &quot;blupADC&quot;)  #  path of example files 
phased=geno_format(
         input_data_path=data_path,      # input data path 
         input_data_name=&quot;example.vcf&quot;,  # input data name,for vcf data
         input_data_type=&quot;VCF&quot;,          # input data type
         phased_genotype=TRUE,           # whether the vcf data has been phased
         haplotype_window_nSNP=5,        # according to nSNP define haplotype-block,
         bigmemory_cal=TRUE,             # format conversion via bigmemory object
         bigmemory_data_path=getwd(),    # path of bigmemory data 
         bigmemory_data_name=&quot;test_blupADC&quot;, #name of bigmemory data 
         output_data_type=c(&quot;Haplotype&quot;,&quot;Numeric&quot;),# output data format
         return_result=TRUE,             #save result in R environment
         cpu_cores=1                     # number of cpu 
                  )</code></pre>
</div>
</div>
<div id="output" class="section level2">
<h2>Output</h2>
<p>According to the result of output, we find that the output contains 5 parts, including:</p>
<ul>
<li><strong>hmp </strong> : <code>Hapmap</code> format genotype data</li>
</ul>
<p>The first column stands for the name of SNP, the thrid column stands for chromosome, the fourth column stands for the physical postion, and the twelth column and the after columns stand for the genotype data</p>
<table style="width:100%;">
<colgroup>
<col width="4%" />
<col width="7%" />
<col width="5%" />
<col width="6%" />
<col width="6%" />
<col width="8%" />
<col width="6%" />
<col width="8%" />
<col width="10%" />
<col width="10%" />
<col width="6%" />
<col width="4%" />
<col width="4%" />
<col width="4%" />
<col width="4%" />
</colgroup>
<thead>
<tr class="header">
<th>rs#</th>
<th>alleles</th>
<th>chrom</th>
<th>pos</th>
<th>strand</th>
<th>assembly</th>
<th>center</th>
<th>protLSID</th>
<th>assayLSID</th>
<th>panelLSID</th>
<th>QCcode</th>
<th>3098</th>
<th>3498</th>
<th>3297</th>
<th>2452</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>SNP1</td>
<td>NA</td>
<td>1</td>
<td>224488</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>CC</td>
<td>AC</td>
<td>AC</td>
<td>CC</td>
</tr>
<tr class="even">
<td>SNP2</td>
<td>NA</td>
<td>1</td>
<td>293696</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>GG</td>
<td>TG</td>
<td>TG</td>
<td>GG</td>
</tr>
<tr class="odd">
<td>SNP3</td>
<td>NA</td>
<td>1</td>
<td>333333</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>GG</td>
<td>TT</td>
<td>TT</td>
<td>GG</td>
</tr>
<tr class="even">
<td>SNP4</td>
<td>NA</td>
<td>1</td>
<td>464830</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>CC</td>
<td>CC</td>
<td>CC</td>
<td>CC</td>
</tr>
<tr class="odd">
<td>SNP5</td>
<td>NA</td>
<td>1</td>
<td>722623</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>AA</td>
<td>GG</td>
<td>GG</td>
<td>AA</td>
</tr>
<tr class="even">
<td>SNP6</td>
<td>NA</td>
<td>1</td>
<td>838596</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>CC</td>
<td>TC</td>
<td>TT</td>
<td>CC</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>ped</strong> : <code>Plink</code> format ped data</li>
</ul>
<p>The first column stands for family name,the second column stands for the individual name,the seventh column and the after columns stand for the genotype data</p>
<table>
<thead>
<tr class="header">
<th>3098</th>
<th>3098</th>
<th>0</th>
<th>0</th>
<th>0</th>
<th>0</th>
<th>C</th>
<th>C</th>
<th>G</th>
<th>G</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>3498</td>
<td>3498</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>A</td>
<td>C</td>
<td>T</td>
<td>G</td>
</tr>
<tr class="even">
<td>3297</td>
<td>3297</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>A</td>
<td>C</td>
<td>T</td>
<td>G</td>
</tr>
<tr class="odd">
<td>2452</td>
<td>2452</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>C</td>
<td>C</td>
<td>G</td>
<td>G</td>
</tr>
<tr class="even">
<td>4255</td>
<td>4255</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>A</td>
<td>C</td>
<td>G</td>
<td>G</td>
</tr>
<tr class="odd">
<td>2946</td>
<td>2946</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>C</td>
<td>C</td>
<td>T</td>
<td>G</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>map </strong>: <code>Plink</code>format map data</li>
</ul>
<p>The first column stands for chromosome, the second column stands for the name of SNP, the thrid column stands for the genetic positon(CM), and the fourth column stands for the physical position</p>
<table>
<thead>
<tr class="header">
<th>1</th>
<th>SNP1</th>
<th>0.224488</th>
<th>224488</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>SNP2</td>
<td>0.293696</td>
<td>293696</td>
</tr>
<tr class="even">
<td>1</td>
<td>SNP3</td>
<td>0.333333</td>
<td>333333</td>
</tr>
<tr class="odd">
<td>1</td>
<td>SNP4</td>
<td>0.464830</td>
<td>464830</td>
</tr>
<tr class="even">
<td>1</td>
<td>SNP5</td>
<td>0.722623</td>
<td>722623</td>
</tr>
<tr class="odd">
<td>1</td>
<td>SNP6</td>
<td>0.838596</td>
<td>838596</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>blupf90 </strong>: <code>BLUPF90</code> format genotype data</li>
</ul>
<p>The first column stands for individual name, the second column stands for the genotype data(numeric)</p>
<table>
<thead>
<tr class="header">
<th>3098</th>
<th>200000</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>3498</td>
<td>112021</td>
</tr>
<tr class="even">
<td>3297</td>
<td>112022</td>
</tr>
<tr class="odd">
<td>2452</td>
<td>200000</td>
</tr>
<tr class="even">
<td>4255</td>
<td>102011</td>
</tr>
<tr class="odd">
<td>2946</td>
<td>212000</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>numeric </strong>: <code>Numeric</code> format genotype data</li>
</ul>
<p>rownames of numeric data stands for the individual name, colnames of numeric data stands for the name of SNP, 0,1,2 stand for the numeric genotype</p>
<table>
<thead>
<tr class="header">
<th>2</th>
<th>0</th>
<th>0</th>
<th>0</th>
<th>0</th>
<th>0</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>1</td>
<td>2</td>
<td>0</td>
<td>2</td>
<td>1</td>
</tr>
<tr class="even">
<td>1</td>
<td>1</td>
<td>2</td>
<td>0</td>
<td>2</td>
<td>2</td>
</tr>
<tr class="odd">
<td>2</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>1</td>
<td>0</td>
<td>2</td>
<td>0</td>
<td>1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>2</td>
<td>1</td>
<td>2</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
</tbody>
</table>
<ul>
<li><p><strong>haplotype_hap</strong>: <code>Haplotype</code> format genotype data.</p>
<p>Row stands for marker, column stands for individual, each individual has two columns;</p>
<table>
<thead>
<tr class="header">
<th>0</th>
<th>0</th>
<th>0</th>
<th>1</th>
<th>1</th>
<th>0</th>
<th>0</th>
<th>0</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>1</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>0</td>
<td>0</td>
</tr>
</tbody>
</table></li>
<li><p><strong>haplotype_sample</strong>: <code>Haplotype</code> format genotype data</p>
<table>
<thead>
<tr class="header">
<th>3098</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>3498</td>
</tr>
<tr class="even">
<td>3297</td>
</tr>
<tr class="odd">
<td>2452</td>
</tr>
<tr class="even">
<td>4255</td>
</tr>
<tr class="odd">
<td>2946</td>
</tr>
</tbody>
</table></li>
<li><p><strong>haplotype_map</strong>: <code>Haplotype</code> format genotype data</p>
<table>
<thead>
<tr class="header">
<th>1</th>
<th>SNP1</th>
<th>224488</th>
<th>C</th>
<th>A</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>SNP2</td>
<td>293696</td>
<td>G</td>
<td>T</td>
</tr>
<tr class="even">
<td>1</td>
<td>SNP3</td>
<td>333333</td>
<td>T</td>
<td>G</td>
</tr>
<tr class="odd">
<td>1</td>
<td>SNP4</td>
<td>464830</td>
<td>A</td>
<td>G</td>
</tr>
<tr class="even">
<td>1</td>
<td>SNP5</td>
<td>722623</td>
<td>C</td>
<td>T</td>
</tr>
<tr class="odd">
<td>1</td>
<td>SNP6</td>
<td>838596</td>
<td>C</td>
<td>A</td>
</tr>
</tbody>
</table></li>
<li><p><strong>vcf </strong>: <code>VCF</code> format genotype data</p></li>
</ul>
<table>
<thead>
<tr class="header">
<th>##fileformat=VCFv4.2</th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>##source=“beagle.29May21.d6d.jar”</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>##INFO&lt;ID=AF,Number=A,Type=Float&gt;</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>##INFO&lt;ID=IMP,Number=0,Type=Flag"&gt;</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>##FORMAT&lt;ID=GT,Number=1,Type=String&gt;</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>#CHROM</td>
<td>POS</td>
<td>ID</td>
<td>REF</td>
<td>ALT</td>
<td>QUAL</td>
<td>FILTER</td>
<td>INFO</td>
<td>FORMAT</td>
<td>3498</td>
<td>3297</td>
</tr>
<tr class="even">
<td>1</td>
<td>6260</td>
<td>M2</td>
<td>T</td>
<td>A</td>
<td>.</td>
<td>PASS</td>
<td>.</td>
<td>GT</td>
<td>1|0</td>
<td>0|1</td>
</tr>
<tr class="odd">
<td>1</td>
<td>15289</td>
<td>M17</td>
<td>A</td>
<td>T</td>
<td>.</td>
<td>PASS</td>
<td>.</td>
<td>GT</td>
<td>0|0</td>
<td>0|0</td>
</tr>
</tbody>
</table>
</div>
<div id="parameter" class="section level2">
<h2>Parameter</h2>
<div id="basic" class="section level3">
<h3>🤡Basic</h3>
<ul>
<li><strong>1: input_data_plink_ped</strong></li>
</ul>
<p>User-provided <code>Plink</code>-ped format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<ul>
<li><strong>2:input_data_plink_map</strong></li>
</ul>
<p>User-provided <code>Plink</code>-map format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<ul>
<li><strong>3:input_data_hmp</strong></li>
</ul>
<p>User-provided <code>Hapmap</code> format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<ul>
<li><strong>4:input_data_BLUPF90</strong></li>
</ul>
<p>User-provided <code>BLUPF90</code> format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<ul>
<li><strong>5:input_data_numeric</strong></li>
</ul>
<p>User-provided <code>Numeric</code> format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<ul>
<li><strong>6:input_data_haplotype_hap</strong></li>
</ul>
<p>User-provided <code>Haplotype</code> format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<ul>
<li><strong>7:input_data_haplotype_sample</strong></li>
</ul>
<p>User-provided <code>Haplotype</code> format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<ul>
<li><strong>8:input_data_haplotype_map</strong></li>
</ul>
<p>User-provided <code>Haplotype</code> format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<ul>
<li><strong>9:input_data_vcf</strong></li>
</ul>
<p>User-provided <code>VCF</code> format genotype data ，<code>data.frame</code> or <code>matrix</code> class.</p>
<p><strong>Note:</strong> <code>input_data_numeric</code> should contain both rownames and colnames.</p>
<p><strong>In addition, for convenience, users can provide the file name, file path, and file type of genotype data directly without reading them in R environment.</strong></p>
<ul>
<li><p><strong>10:input_data_type</strong></p>
<p>File type of provided genotype data ，<code>character</code> class.</p>
<ul>
<li>Hapmap</li>
<li>Plink</li>
<li>BLUPF90</li>
<li>Numeric</li>
<li>Haplotype</li>
<li>VCF</li>
</ul></li>
<li><p><strong>11:input_data_path</strong></p></li>
</ul>
<p>File path of provided genotype data ，<code>character</code> class.</p>
<ul>
<li><strong>12:input_data_name</strong></li>
</ul>
<p>File name of provided genotype data ，<code>character</code> class.</p>
<p><strong>Note:</strong> if input_data_type is Plink or VCF, user don’t need to include suffix in the file name of genotype data.</p>
<p>eg. for Plink type data, files name are test1.map and test1.ped，we should set<code>input_data_name="test1"</code>.</p>
<ul>
<li><strong>13:output_data_name</strong></li>
</ul>
<p>File name of output genotype data, <code>character</code> class.</p>
<ul>
<li><p><strong>14:output_data_type</strong></p>
<p>File type of output genotype data, <code>character</code> class.</p>
<ul>
<li>Hapmap</li>
<li>Plink</li>
<li>BLUPF90</li>
<li>Numeric</li>
<li>Haplotype</li>
<li>VCF</li>
</ul></li>
</ul>
<p><strong>Note</strong>: users can output multiple formats of genotype data simultaneously. e.g.<code>output_data_type=c("Hapmap","Plink","BLUPF90","Numeric")</code>, outout 4 types of genotype data simultaneously .</p>
<ul>
<li><strong>15:return_result</strong></li>
</ul>
<p>Whether return result, <code>logical</code> class. Default is FALSE.</p>
<p>Additionally, for convenience, users can save output genotype data into local computer .</p>
<ul>
<li><strong>16:bigmemory_cal</strong></li>
</ul>
<p>Whether using bigmemory method to calculate. <code>logical</code> class. Default is FALSE.</p>
<ul>
<li><strong>17:bigmemory_data_path</strong></li>
</ul>
<p>The file path bigmemory data . <code>character</code> class.</p>
<ul>
<li><strong>18:bigmemory_data_name</strong></li>
</ul>
<p>The file name bigmemory data . <code>character</code> class.</p>
<ul>
<li><strong>19:phased_genotype</strong></li>
</ul>
<p>Whether genotype data has been phased. <code>logical</code> class. Default is FALSE.</p>
<ul>
<li><strong>20:haplotype_window_nSNP</strong></li>
</ul>
<p>According to the number of consecutive SNPs define haplotype block. <code>numeric</code> class. Default is NULL.</p>
<ul>
<li><strong>21:haplotype_window_kb</strong></li>
</ul>
<p>According to the physical location define haplotype block. <code>numeric</code> class. Default is NULL.</p>
<ul>
<li><strong>22:haplotype_window_block</strong></li>
</ul>
<p>According to user-provided block to define haplotype block . <code>numeric</code> class. Default is NULL.</p>
<p>The first column is the position of window start, the second column is the position of window end.</p>
<table>
<thead>
<tr class="header">
<th>1</th>
<th>5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>6</td>
<td>10</td>
</tr>
<tr class="even">
<td>11</td>
<td>15</td>
</tr>
<tr class="odd">
<td>16</td>
<td>20</td>
</tr>
<tr class="even">
<td>21</td>
<td>25</td>
</tr>
<tr class="odd">
<td>26</td>
<td>30</td>
</tr>
</tbody>
</table>
</div>
<div id="advanced" class="section level3">
<h3>💨Advanced</h3>
<ul>
<li><strong>23:cpu_cores</strong></li>
</ul>
<p>Number of cpu in calculating, <code>numeric</code> class. Default is 1.</p>
<ul>
<li><strong>24:miss_base</strong></li>
</ul>
<p>Missing genotype character, <code>character</code> class. Default is “NN”.</p>
<ul>
<li><strong>25:miss_base_num</strong></li>
</ul>
<p>Missing genotype number after numeric conversion, <code>numeric</code> class. Default is 5.</p>
<ul>
<li><strong>26:miss_base_num</strong></li>
</ul>
<p>Missing genotype number after numeric conversion, <code>numeric</code> class. Default is 5.</p>
</div>
</div>
