---
title: 'R package:blupADC-ç®€ä»‹'
author: qsmei
date: '2021-04-13'
slug: overView
categories:
  - blupADC
tags: []
subtitle: ''
summary: 'ç®€ä»‹'
authors: []
lastmod: '2021-04-13T22:32:36+08:00'
featured: no
image:
  caption: ''
  focal_point: ''
  preview_only: no
projects: []
toc: true
---

{{<toc>}}<img src="https://qsmei-markdown.oss-cn-shanghai.aliyuncs.com/markdown-img/20210617165506.png" alt="logo-blupADC"  height="250" align="right" style="zoom:25%;" />

**Documents support two-language([English](https://qsmei.netlify.app/post/feature-0-overview/overview/) and [Chinese](https://qsmei.netlify.app/zh/post/feature-0-overview/overview/)).** 

## ç®€ä»‹

`blupADC` æ˜¯ä¸€ä¸ªä¸“æ³¨äºåˆ†æåŠ¨æ¤ç‰©è‚²ç§ä¸­çš„ç³»è°±æ•°æ®ã€åŸºå› å‹æ•°æ®åŠé—ä¼ è¯„ä¼°çš„å·¥å…·ã€‚åœ¨è®¾è®¡è¯¥å·¥å…·æ—¶ï¼Œæˆ‘ä»¬å¯¹æ•°æ®å¤„ç†æ—¶å¯èƒ½é‡åˆ°çš„å„ç§é—®é¢˜å‡è¿›è¡Œäº†è¯¦ç»†çš„è€ƒé‡(**ps.å¦‚æœæ‚¨æœ‰å¥½çš„å»ºè®®ï¼Œè¯·ç§¯æè”ç³»ä½œè€…!**)ã€‚æ­¤å¤–ï¼Œä¸ºäº†æé«˜åˆ†ææ•ˆç‡ï¼Œ `blupADC` å¯æ”¯æŒå¹¶è¡Œè®¡ç®—(é€šè¿‡`openMP` ) åŠå¤§æ•°æ®å¤„ç†(é€šè¿‡`bigmemory`)ï¼Œå¹¶ä¸”`blupADC` ä¸­çš„æ ¸å¿ƒå‡½æ•°å‡é€šè¿‡c++(`Rcpp` and `RcppArmadillo `) è¿›è¡Œç¼–å†™ã€‚

`blupADC`æä¾›äº†è®¸å¤šæœ‰ç”¨çš„åŠŸèƒ½åœ¨æ•´ä¸ªåŠ¨æ¤ç‰©è‚²ç§çš„æµç¨‹ä¸­ï¼ŒåŒ…æ‹¬ ç³»è°±åˆ†æ(ç³»è°±è¿½æº¯ã€é‡å‘½ååŠçº é”™)ï¼ŒåŸºå› å‹æ•°æ®æ ¼å¼è½¬æ¢(æ”¯æŒ**Hapmap, Plink, BLUPF90, Numeric, Haplotype åŠVCF** æ ¼å¼)ï¼ŒåŸºå› å‹æ•°æ®çš„è´¨æ§å¡«å……ï¼Œäº²ç¼˜å…³ç³»çŸ©é˜µçš„æ„å»º(**ç³»è°±ï¼ŒåŸºå› ç»„åŠä¸€æ­¥æ³•äº²ç¼˜å…³ç³»çŸ©é˜µ**)ä»¥åŠé—ä¼ è¯„ä¼°ï¼ˆä»…éœ€å‡ è¡Œä»£ç å³å¯é€šè¿‡DMUå’ŒBLUPF90å®Œæˆé—ä¼ è¯„ä¼°ï¼‰ã€‚

æœ€åï¼Œä¸ºäº†æ–¹è¿›ä¸€æ­¥æ–¹ä¾¿ç”¨æˆ·çš„ä½¿ç”¨(å°¤å…¶æ˜¯ç¼–ç¨‹åŸºç¡€å¼±çš„ç”¨æˆ·)ï¼Œæˆ‘ä»¬åˆ›å»ºäº†ä¸€ä¸ªå…è´¹çš„åœ¨çº¿ç½‘ç«™([shinyapp](http://47.95.251.15:443/blupADC/))ã€‚ç›¸å…³çš„åŠŸèƒ½ä»åœ¨å¼€å‘ä¸­ã€‚ä½†æ˜¯ï¼Œç½‘ç«™çš„ä¸€ä¸ªç¼ºç‚¹å°±æ˜¯ï¼Œä¸èƒ½å¤„ç†å¤§æ•°æ®ï¼Œè¯·å¤§å®¶åˆç†é€‰æ‹©ï¼

ğŸ˜Š ç¥å¥½è¿ !   å¦‚æœä½ æœ‰å»ºè®®æˆ–è€…é—®é¢˜ï¼Œè¯·è”ç³»: [hzau_qsmei@163.com](mailto:hzau_qsmei@163.com) !

## æ–°æ·»åŠ çš„åŠŸèƒ½ 

### 1.0.3

- ç›®å‰èƒ½å¤Ÿé€šè¿‡DMUè‡ªåŠ¨åˆ†æ æ¯æ€§æ•ˆåº”ï¼Œæ°¸ä¹…ç¯å¢ƒæ•ˆåº”ï¼Œéšæœºå›å½’æ•ˆåº” åŠ ç¤¾ä¼šé—ä¼ æ•ˆåº”çš„æ¨¡å‹ (2021.8.24)

### 1.0.4

- æ”¯æŒå•å€å‹æ ¼å¼è½¬æ¢ï¼Œå•å€å‹-æ•°å­—çŸ©é˜µåŠå•å€å‹åŠ æ€§äº²ç¼˜å…³ç³»çŸ©é˜µçš„æ„å»º(2021.10.8)
- å¼•å…¥bigmemoryå¯¹è±¡æ”¯æŒå¤§æ•°æ®åˆ†æ(2021.10.8)

### 1.1.0 

- å¼•å…¥é¢å‘å¯¹è±¡çš„æ–¹å¼è¿›è¡ŒåŸºå› ç»„é¢„æµ‹ï¼ˆé€‰æ‹©)åˆ†æ (2023.07.17) ([see more details](https://qsmei.netlify.app/post/r6-genomic-predictionselection/blup/)) 
- å°†æ•°æ®å’Œè½¯ä»¶å‡è½¬ç§»è‡³, `blupSUP`, ç”¨æˆ·é¦–æ¬¡è¿è¡Œæ–°ç‰ˆæœ¬å‰éœ€å®‰è£…`blupSUP`!
- ç”¨æˆ·ä»ç„¶å¯ä½¿ç”¨ä¹‹å‰çš„ä»£ç è¿è¡ŒblupADCï¼

## å¼€å§‹

### ğŸ™Šå®‰è£…

 å®‰è£…`blupADC` ä¹‹å‰ï¼Œç”¨æˆ·é¦–å…ˆéœ€è¦å®‰è£…å¦‚ä¸‹5ä¸ªåŒ…ï¼š`Rcpp`, `RcppArmadillo` and `data.table`.

```R
install.packages(c("Rcpp", "RcppArmadillo","RcppProgress","data.table","bigmemory"))
```

**ğŸ‘‰ Note:  åœ¨ DMU å’Œ BLUPF90çš„åˆ†æä¸­ï¼Œæˆ‘ä»¬é€šå¸¸éœ€è¦æå‰ä¸‹è½½å¥½ DMU è½¯ä»¶ ([DMUä¸‹è½½ç½‘ç«™](https://dmu.ghpc.au.dk/dmu/))  å’Œ BLUPF90è½¯ä»¶ ([BLUPF90ä¸‹è½½ç½‘ç«™](http://nce.ads.uga.edu/html/projects/programs/))ã€‚ä¸ºäº†æ–¹ä¾¿ç”¨æˆ·ä½¿ç”¨ï¼Œæˆ‘ä»¬å·²ç»å°†ä¸¤æ¬¾è½¯ä»¶ä¸­åŸºç¡€æ¨¡å—å°è£…è¿›äº† `blupADC`ï¼Œè¯·å¤§å®¶åˆç†ä½¿ç”¨ã€‚**

**å¦‚æœæ‚¨æƒ³å°†DMUå’ŒBLUPF90ç”¨ä½œå•†ä¸šç”¨é€”ï¼Œè¯·åŠ¡å¿…è”ç³» DMU å’Œ BLUPF90çš„ä½œè€…ï¼ï¼ï¼** 

#### é¦–å…ˆè¿è¡Œæ–°ç‰ˆæœ¬çš„`blupADC`,ç”¨æˆ·éœ€å®‰è£…`blupSUP`ï¼ˆä»…éœ€å®‰è£…ä¸€æ¬¡)

```R
devtools::install_github("TXiang-lab/blupSUP")
```

#### Install blupADC via devtools

```R
devtools::install_github("TXiang-lab/blupADC")
```

ğŸ‘‰ **Note:If the connection with github is not good(such as in China), user can download as below:**  

```R
devtools::install_git("https://gitee.com/qsmei/blupADC")
```

å®‰è£…æˆåŠŸåï¼Œæˆ‘ä»¬è¾“å…¥å¦‚ä¸‹ä»£ç å³å¯åŠ è½½RåŒ…:

``` {.r}
library(blupADC)
```

### ğŸ™ŠåŠŸèƒ½

-   åŠŸèƒ½ 1. åŸºå› å‹æ•°æ®é—´çš„æ ¼å¼è½¬æ¢
-   åŠŸèƒ½ 2. åŸºå› å‹æ•°æ®çš„è´¨æ§ä¸å¡«å……
-   åŠŸèƒ½ 3. å“ç§åˆ†æåŠåŸºå› å‹æ•°æ®é‡å¤æ€§æ£€æµ‹
-   åŠŸèƒ½ 4. ç³»è°±è¿½æº¯ã€é‡å‘½ååŠçº é”™
-   åŠŸèƒ½ 5. ç³»è°±å¯è§†åŒ–
-   åŠŸèƒ½ 6. äº²ç¼˜å…³ç³»çŸ©é˜µçš„æ„å»º(A,G, H)
-   åŠŸèƒ½ 7. åˆ©ç”¨DMUè½¯ä»¶è¿›è¡Œé—ä¼ è¯„ä¼°
-   åŠŸèƒ½ 8. åˆ©ç”¨BLUPF90è½¯ä»¶è¿›è¡Œé—ä¼ è¯„ä¼°

## ä½¿ç”¨

`blupADC` å†…ç½®äº†å‡ ä¸ªæ•°æ®é›†å¯¹è±¡, åŒ…æ‹¬ `data_hmp` åŠ`origin_pedigree`.

æ­¤å¤–ï¼Œ`blupSUP`  æä¾›ä¸€äº›ç¤ºä¾‹æ–‡ä»¶ï¼Œ è¿™äº›æ–‡ä»¶å­˜å‚¨åœ¨`~/blupSUP/extdata` è·¯å¾„ä¸‹ã€‚é€šè¿‡è¾“å‡ºä¸‹é¢çš„ä»£ç ï¼Œæˆ‘ä»¬å°±èƒ½å¾—åˆ° è¿™äº›æ–‡ä»¶çš„ç»å¯¹è·¯å¾„äº†ï¼š

``` {.r}
system.file("extdata", package = "blupSUP") # path of provided files
```

#### åŠŸèƒ½ 1. åŸºå› å‹æ•°æ®é—´çš„æ ¼å¼è½¬æ¢ ([see more details](https://qsmei.netlify.app/zh/post/blupadc/))

``` R
library(blupADC)
format_result=geno_format(
    	input_data_hmp=example_data_hmp,  # provided data variable
        output_data_type=c("Plink","BLUPF90","Numeric"),# output data format
    	output_data_path=getwd(),   #output data path      
    	output_data_name="blupADC", #output data name    
        return_result = TRUE,       #save result in R environment
        cpu_cores=1                 # number of cpu 
                  )

#convert phased VCF data to haplotype format and  haplotype-based numeric format
library(blupADC)
data_path=system.file("extdata", package = "blupSUP")  #  path of example files 
phased=geno_format(
         input_data_path=data_path,      # input data path 
         input_data_name="example.vcf",  # input data name,for vcf data
         input_data_type="VCF",          # input data type
         phased_genotype=TRUE,           # whether the vcf data has been phased
         haplotype_window_nSNP=5,        # according to nSNP define haplotype-block,
    	 bigmemory_cal=TRUE,             # format conversion via bigmemory object
    	 bigmemory_data_path=getwd(),    # path of bigmemory data 
    	 bigmemory_data_name="test_blupADC", #name of bigmemory data 
         output_data_type=c("Haplotype","Numeric"),# output data format
         return_result=TRUE,             #save result in R environment
         cpu_cores=1                     # number of cpu 
                  )
```

#### åŠŸèƒ½ 2. åŸºå› å‹æ•°æ®çš„è´¨æ§ä¸å¡«å…… ([see more details](https://qsmei.netlify.app/zh/post/feature-2-qc_imputation/qc_imputation/))

``` R
library(blupADC)
geno_qc_impute(
            input_data_hmp=example_data_hmp,        #provided data variable
            data_analysis_method="QC_Imputation",   #analysis method type,QC + imputatoin
            output_data_path=getwd(),               #output data path
            output_data_name="YY_data",             #output data name
            output_data_type="VCF"                  #output data format 
            )                       
```

#### åŠŸèƒ½ 3. å“ç§åˆ†æåŠåŸºå› å‹æ•°æ®é‡å¤æ€§æ£€æµ‹ ([see more details](https://qsmei.netlify.app/zh/post/feature-3-overlap_pca/blupadc/))

``` R
library(blupADC)
check_result=geno_check(
                  input_data_hmp=example_PCA_data_hmp,   #provided hapmap data object
                  duplication_check=FALSE,       #whether check the duplication of genotype
                  breed_check=TRUE,               # whether check the record of breed
                  breed_record=example_PCA_Breed, # provided breed record
                  output_data_path=getwd(),       #output path
                  return_result=TRUE              #save result as a R environment variable
                  )
```

#### åŠŸèƒ½ 4. ç³»è°±è¿½æº¯ã€é‡å‘½ååŠçº é”™ ([see more details](https://qsmei.netlify.app/zh/post/feature-4-trace_pedigree/pedigree/))

``` R
library(blupADC)
pedigree_result=trace_pedigree(
                input_pedigree=example_ped1,   #provided pedigree data variable
                trace_generation=3,            # trace generation
                output_pedigree_tree=T         # output pedigree tree
                )  
```

#### åŠŸèƒ½ 5. ç³»è°±å¯è§†åŒ– ([see more details](https://qsmei.netlify.app/zh/post/feature-5-visualize_pedigree/pedigree/))

``` R
library(blupADC)
plot=ggped(
       input_pedigree=example_ped2,
       trace_id=c("121"),
       trace_sibs=TRUE   #whether plot the sibs of subset-id  
        ) 
```

#### åŠŸèƒ½ 6. äº²ç¼˜å…³ç³»çŸ©é˜µçš„æ„å»º(A,G, H)  ([see more details](https://qsmei.netlify.app/zh/post/feature-6-kinship_matrix/relationship_matrix/))

``` R
library(blupADC)
data_path=system.file("extdata", package = "blupSUP")  #  path of example files 
kinship_result=cal_kinship(
        		input_data_path=data_path,      # input data path 
        		input_data_name="example.vcf",  # input data name,for vcf data
         		input_data_type="VCF",          # input data type
    			kinship_type=c("G_A","G_D"),      #type of  kinship matrix
    			dominance_type=c("genotypic"),    #type of dominance effect
    			inbred_type=c("Homozygous"),      #type of inbreeding coefficients
    			return_result=TRUE)               #save result as a R environment variable                
```

#### åŠŸèƒ½ 7. åˆ©ç”¨DMUè½¯ä»¶è¿›è¡Œé—ä¼ è¯„ä¼° ([see more details](https://qsmei.netlify.app/zh/post/feature-7-run_dmu/run_dmu/))

``` R
library(blupADC)
data_path=system.file("extdata", package = "blupSUP")  #  path of example files 
  
run_DMU(
        phe_col_names=c("Id","Mean","Sex","Herd_Year_Season","Litter","Trait1","Trait2","Age"), # colnames of phenotype 
        target_trait_name=list(c("Trait1")),                     #trait name 
        fixed_effect_name=list(c("Sex","Herd_Year_Season")),     #fixed effect name
        random_effect_name=list(c("Id","Litter")),               #random effect name
        covariate_effect_name=NULL,                              #covariate effect name
        phe_path=data_path,                          #path of phenotype file
        phe_name="phenotype.txt",                    #name of phenotype file
        integer_n=5,                                 #number of integer variable 
        analysis_model="PBLUP_A",                    #model of genetic evaluation
        dmu_module="dmuai",                          #modeule of estimating variance components 
        relationship_path=data_path,                 #path of relationship file 
        relationship_name="pedigree.txt",            #name of relationship file 
        output_result_path=getwd()                   # output path 
        )
```

#### åŠŸèƒ½ 8. åˆ©ç”¨BLUPF90è½¯ä»¶è¿›è¡Œé—ä¼ è¯„ä¼° ([see more details](https://qsmei.netlify.app/zh/post/feature-8-run_blupf90/blupf90/))

``` R
library(blupADC)
data_path=system.file("extdata", package = "blupSUP")  #  path of example files 
  
run_BLUPF90(
        phe_col_names=c("Id","Mean","Sex","Herd_Year_Season","Litter","Trait1","Trait2","Age"), # colnames of phenotype 
        target_trait_name=list(c("Trait1")),                     #trait name 
        fixed_effect_name=list(c("Sex","Herd_Year_Season")),     #fixed effect name
        random_effect_name=list(c("Id","Litter")),               #random effect name
        covariate_effect_name=NULL,                              #covariate effect name
        phe_path=data_path,                          #path of phenotype file
        phe_name="phenotype.txt",                    #name of phenotype file
        analysis_model="PBLUP_A",                    #model of genetic evaluation
        relationship_path=data_path,                 #path of relationship file 
        relationship_name="pedigree.txt",            #name of relationship file 
        output_result_path=getwd()                   # output path 
        )
```
