---
title: 'R package:blupADC-Feature 3'
author: qsmei
date: '2021-04-16'
slug: blupADC
categories: ["Check genotype"]
tags: []
subtitle: ''
summary: 'Breed composition analysis '
authors: []
lastmod: '2021-04-15T15:11:30+08:00'
featured: no
image:
  caption: ''
  focal_point: ''
  preview_only: no
projects: []
toc: true
---

{{<toc>}} 

## Overview   

ðŸ‘¦   Breed composition analysis is usually a problem in data analysis. In package:`blupADC`, user can solve this problem by applying `genotype_data_check` function. In addition, user can  detect the duplication of genomic data easily by applying `genotype_data_check` function.

## Example 

### Breed composition analysis 

``` {.R}
library(blupADC)
check_result=genotype_data_check(
                  input_data_hmp=PCA_data_hmp,   #provided hapmap data object
                  duplication_check=FALSE,       #whether check the duplication of genotype
                  breed_check=TRUE,              # whether check the record of breed
                  breed_record=PCA_Breed,        # provided breed record
                  return_result=TRUE             #return result 
                  )
```

### Check duplication 

``` {.R}
library(blupADC)
check_result=genotype_data_check(
                  input_data_hmp=data_hmp,   #provided hapmap data object
                 duplication_threshold=0.95, #threshold of duplication
                  duplication_check=TRUE,    #whether check the duplication of genotype
                  breed_check=FALSE,         # whether check the record of breed
                  return_result=TRUE         #return result 
                  )
```

## Output

```{r, echo=FALSE, message=FALSE, warning=FALSE}
library(blupADC)
library(ggplot2)
check_result=genotype_data_check(
                  input_data_hmp=PCA_data_hmp,   #provided hapmap data object
                  duplication_check=FALSE,        #whether check the duplication of genotype
                  breed_check=TRUE,              # whether check the record of breed
                  breed_record=PCA_Breed,        # provided breed record
                  return_result=TRUE             #return result 
                  )
pca_outlier=check_result$breed$outlier
rownames(pca_outlier)=NULL
check_result=genotype_data_check(
                  input_data_hmp=cbind(PCA_data_hmp,PCA_data_hmp[,12:15]),   #provided hapmap data object
                  duplication_check=TRUE,        #whether check the duplication of genotype
                  breed_check=FALSE,              # whether check the record of breed
                  breed_record=PCA_Breed,         # provided breed record
                  return_result=TRUE              #return result 
                  )
duplicated_genotype=check_result$duplicate_genotype
```

The result of output mainly contains two parts, including:

- **duplicated_genotype**    

```{r input_data_hmp3,echo=FALSE}
knitr::kable(head(duplicated_genotype))
```

The first and the second column is the name of individual, the third column is the percentage of overlap.

- **pca_outlier** 

```{r input_data_hmp2,echo=FALSE}
knitr::kable(head(pca_outlier))
```

Figure A is the  PCA result before correcting breed record  , Figure B is the PCA result after correcting breed correcting record

![PCA_blupADC](https://qsmei-markdown.oss-cn-shanghai.aliyuncs.com/markdown-img/20210805034645.png)

## Parameter 

Many parameters in  `genotype_data_overlap` are the same as in `genotype_data_format_conversion` function ([see more details](https://qsmei.netlify.app/post/blupadc/)). 

Thus, we will introduce  specific parameters in  `genotype_data_overlap`  function.

-   **1ï¼šselected_snps** 

Number of SNPs in detecting overlap, `numeric` class. Default is 1000.

-   **2ï¼šoverlap_threshold**

Threshold of duplicate genotype, `numeric` class. Default is 0.95.

-   **3ï¼šduplication_check**

Whether check duplication of genotype, `logical` class. Default is TRUE.

-   **4ï¼šbreed_check**

Whether check breed record of genotype, `logical` class. Default is FALSE.

-   **5ï¼šind_breed**

Breed record of individuals, `data.frame` class.  

The format of `ind_breed` is showing as follow:

```{r, echo=FALSE, message=FALSE, warning=FALSE}
knitr::kable(head(blupADC::PCA_Breed))
```

When the proportion of genotype data between two individuals is larger than this threshold, these two individuals will be regarded as the same individual.


